---
title: "Cortical cells DGE E13.5 Data-set Anaysis"
---

```{r, include=FALSE}
library(ggplot2)
library(tibble)
library(zeallot)
library(COTAN)

options(parallelly.fork.enable = TRUE)

outDir <- "Data/Yuzwa_MouseCortex/"

setLoggingLevel(1)
setLoggingFile(file.path(outDir, "CorticalCells_GSM2861511_E135-analysis.log"))
```

Read the dataset from file

```{r}
dataset <- read.csv(file.path("Data/Yuzwa_MouseCortex/", "GSM2861511_E135_Only_Cortical_Cells_DGE.txt.gz"),
                    header = TRUE, sep = "\t", strip.white = TRUE,
                    row.names = 1)
```

Crate the COTAN object

```{r}
sampleCondition <- "CorticalCells_GSM2861511_E135"
cc135Obj <- COTAN(raw = dataset)
cc135Obj <- initializeMetaDataset(cc135Obj,
                                  GEO = "GSM2861511_E135",
                                  sequencingMethod = "Drop_seq",
                                  sampleCondition = sampleCondition)
```

Inspect cells' sizes

```{r}
cellSizePlot(cc135Obj)
```

Drop cells with too many ritz reads as they are probably duplets

```{r}
cellsSizeThr <- 7500
cc135Obj <- addElementToMetaDataset(cc135Obj, "Cells size threshold", cellsSizeThr)

cells_to_rem <- getCells(cc135Obj)[getCellsSize(cc135Obj) > cellsSizeThr]
cc135Obj <- dropGenesCells(cc135Obj, cells = cells_to_rem)

cellSizePlot(cc135Obj)
```

Inspect the number of expressed genes per cell

```{r}
genesSizePlot(cc135Obj)
```

Drop cells with too high genes expession as they are probably duplets

```{r}
genesSizeThr <- 3000
cc135Obj <- addElementToMetaDataset(cc135Obj, "Num genes threshold", genesSizeThr)

numExprGenes <- getNumExpressedGenes(cc135Obj)
cells_to_rem <- names(numExprGenes)[numExprGenes > genesSizeThr]
cc135Obj <- dropGenesCells(cc135Obj, cells = cells_to_rem)

genesSizePlot(cc135Obj)
```

Check number of mitocondrial genes expressed in each cell

```{r}
mitGenesPattern <- "^mt-"
getGenes(cc135Obj)[grep(mitGenesPattern, getGenes(cc135Obj))]

c(mitPlot, mitSizes) %<-%
  mitochondrialPercentagePlot(cc135Obj, genePrefix = mitGenesPattern)

plot(mitPlot)
```

Cells with a too high percentage of mitocondrial genes are likely dead (or at the last problematic) cells. So we drop them!

```{r}
mitPercThr <- 10.0
cc135Obj <- addElementToMetaDataset(cc135Obj, "Mitoc. perc. threshold", mitPercThr)

cells_to_rem <- rownames(mitSizes)[mitSizes[["mit.percentage"]] > mitPercThr]

cc135Obj <- dropGenesCells(cc135Obj, cells = cells_to_rem)

c(mitPlot, mitSizes) %<-%
  mitochondrialPercentagePlot(cc135Obj, genePrefix = mitGenesPattern)

plot(mitPlot)
```

Check no further outliers after all the culling

```{r}
cellSizePlot(cc135Obj)
genesSizePlot(cc135Obj)
```

Clean: round 1

```{r}
cc135Obj <- clean(cc135Obj)

c(pcaCellsPlot, pcaCellsData, genesPlot, UDEPlot, nuPlot) %<-% cleanPlots(cc135Obj)

plot(pcaCellsPlot)
plot(genesPlot)
```

B group contains high number of hemoglobin genes: so they are not interesting

```{r}
cells_to_rem <- rownames(pcaCellsData)[pcaCellsData[["groups"]] == "B"]

cc135Obj <- dropGenesCells(cc135Obj, cells = cells_to_rem)
```

Clean: round 2

```{r}
cc135Obj <- clean(cc135Obj)

c(pcaCellsPlot, pcaCellsData, genesPlot, UDEPlot, nuPlot) %<-% cleanPlots(cc135Obj)

plot(pcaCellsPlot)
plot(genesPlot)
```

B group contains just 3 cells quite different in the 3rd-5th components: better to drop them

```{r}
plot(pcaCellsData)

cells_to_rem <- rownames(pcaCellsData)[pcaCellsData[["groups"]] == "B"]

cc135Obj <- dropGenesCells(cc135Obj, cells = cells_to_rem)
```

Clean: round 3

```{r}
cc135Obj <- clean(cc135Obj)

c(pcaCellsPlot, pcaCellsData, genesPlot, UDEPlot, nuPlot) %<-% cleanPlots(cc135Obj)

plot(pcaCellsPlot)
plot(genesPlot)

cc135Obj <- addElementToMetaDataset(cc135Obj, "Num drop B group", 2)
```

Visualize if all is ok:

```{r}
plot(UDEPlot)

plot(nuPlot)

lowUDEThr <- 0.17 # the threshold to remove low UDE cells

nuDf <- data.frame("nu" = sort(getNu(cc135Obj)), "n" = seq_along(getNu(cc135Obj)))
UDEPlot_zoomed <- ggplot(nuDf, aes(x = n, y = nu)) +
            geom_point(colour = "#8491B4B2", size = 1.0) +
            xlim(0L, 400L) +
            ylim(0.0, 1.0) +
            geom_hline(yintercept = lowUDEThr, linetype = "dashed",
                       color = "darkred") +
            annotate(geom = "text", x = 200L, y = 0.25,
                     label = paste0("to remove cells with nu < ", lowUDEThr),
                     color = "darkred", size = 4.5)

plot(UDEPlot_zoomed)
```

Drop very low UDE cells as they are likely outliers

```{r}
cc135Obj <- addElementToMetaDataset(cc135Obj, "Low UDE threshold", lowUDEThr)

cells_to_rem <- rownames(nuDf)[nuDf[["nu"]] < lowUDEThr]
cc135Obj <- dropGenesCells(cc135Obj, cells = cells_to_rem)
```

Final cleaning to check all is OK

```{r}
cc135Obj <- clean(cc135Obj)

c(pcaCellsPlot, pcaCellsData, genesPlot, UDEPlot, nuPlot) %<-% cleanPlots(cc135Obj)

plot(pcaCellsPlot)
plot(genesPlot)
plot(UDEPlot)
plot(nuPlot)

plot(cellSizePlot(cc135Obj))
plot(genesSizePlot(cc135Obj))
```

```{r, eval=FALSE,echo=TRUE}
cc135Obj <- proceedToCoex(cc135Obj, calcCoex = TRUE, cores = 12,
                          saveObj = TRUE, outDir = outDir)
```

Save the COTAN object

```{r, eval=FALSE,echo=TRUE}
saveRDS(cc135Obj, file = file.path(outDir, paste0(sampleCondition, ".cotan.RDS")))
```

```{r, eval=TRUE,echo=FALSE}
cc135Obj <- readRDS("Data/Yuzwa_MouseCortex/CorticalCells_GSM2861511_E135.cotan.RDS")
```


```{r}
gdiData <- calculateGDI(cc135Obj)

genesToLabel <- head(rownames(gdiData[order(gdiData[["GDI"]],
                                            decreasing = TRUE), ]), n = 10L)

genesToLabel

gdiPlot <- GDIPlot(cc135Obj, GDIIn = gdiData, GDIThreshold = 1.4,
                   genes = list("Top 10 GDI genes" = genesToLabel))

plot(gdiPlot)
```


```{r, eval=FALSE, echo=TRUE}
splitClusters <- cellsUniformClustering(cc135Obj, GDIThreshold = 1.4, cores = 13,
                                        saveObj = TRUE, outDir = outDir)

c(splitCoexDF, splitPValueDF) %<-% DEAOnClusters(cc135Obj, clusters = splitClusters)

cc135Obj <- addClusterization(cc135Obj, clName = "split",
                              clusters = splitClusters,
                              coexDF = splitCoexDF, override = TRUE)

table(splitClusters)
```

```{r, eval=FALSE, echo=TRUE}
saveRDS(cc135Obj, file = file.path(outDir, paste0(sampleCondition, ".cotan.RDS")))
```

```{r, eval=FALSE, echo=TRUE}
c(mergedClusters, mergedCoexDF, mergedPValueDF) %<-%
  mergeUniformCellsClusters(cc135Obj, clusters = splitClusters,
                            GDIThreshold = 1.4, cores = 13,
                            saveObj = TRUE, outDir = outDir)

cc135Obj <- addClusterization(cc135Obj, clName = "merge",
                              clusters = mergedClusters,
                              coexDF = mergedCoexDF,
                              override = TRUE)

table(mergedClusters)
```

```{r, eval=FALSE, echo=TRUE}
saveRDS(cc135Obj, file = file.path(outDir, paste0(sampleCondition, ".cotan.RDS")))
```

```{r}
#clustersMarkersHeatmapPlot(cc135Obj, groupMarkers, clName = "merge", kCuts = 4)
```

```{r}
sessionInfo()
```
