<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Marco Fantozzi">
<meta name="dcterms.date" content="2024-01-18">

<title>COTAN Datasets analysis - Mixing Uniform Clusters To Estimate GDI Sensitivity</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./OriginalClustersDistanceEvaluation.html">GDI sensitivity</a></li><li class="breadcrumb-item"><a href="./ForebrainDorsal_MixingUniformClusters_ToEstimateGDISensitivity.html">Mixing Uniform Clusters To Estimate GDI Sensitivity</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">COTAN Datasets analysis</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Home</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false">
 <span class="menu-text">Datasets</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false">
 <span class="menu-text">Mouse Cortex Loo 2019</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Dataset1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cleaning and cell clustering</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneCoexAnalysisDataset1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Coexpression analaysis</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false">
 <span class="menu-text">Mouse Cortex Yuzwa 2017</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./CorticalCells_GSM2861511_E135-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">E13.5 Cleaning and cell clustering</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneCoexAnalysisE13.5Yuzwa.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Correlation analysis for E13.5</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./CorticalCells_GSM2861514_E175-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">E17.5 Cleaning and cell clustering</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneCoexAnalysisE17.5Yuzwa.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Correlation analysis for E17.5</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="false">
 <span class="menu-text">Mouse Brain LaManno 2021</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ForebrainDorsal_E135-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">E13.5 Cleaning and cell clustering</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneCoexAnalysisE13.5MouseBrain.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Correlation analysis for E13.5</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ForebrainDorsal_E150-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">E15.0 Cleaning and cell clustering</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneCoexAnalysisE15.0MouseBrain.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Correlation analysis for E15.0</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ForebrainDorsal_E175-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">E17.5 Cleaning and cell clustering</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneCoexAnalysisE17.5MouseBrain.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Correlation analysis for E17.5</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ForebrainLaManno2021ClustersGDI.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Original Clusters GDI evaluation</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false">
 <span class="menu-text">CD14+</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./cd14_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CD14+ Cleaning and cell clustering</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false">
 <span class="menu-text">PBMCs with antibody-derived tag</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./PBMC1_filtering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Filtering of PBMC1 using COTAN</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./PBMC2_filtering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Filtering of PBMC2 using COTAN</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./PBMC3_filtering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Filtering of PBMC3 using COTAN</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./PBMC4_filtering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Filtering of PBMC4 using COTAN</span></a>
  </div>
</li>
      </ul>
  </li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" aria-expanded="false">
 <span class="menu-text">Coex score evaluation</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-7" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./generalCoexAssess.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">General Coex Evaluation</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" aria-expanded="true">
 <span class="menu-text">GDI sensitivity</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-8" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./OriginalClustersDistanceEvaluation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Clusters Distance Evaluation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./FindUniformOrigCluster.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cluster Unifomity and GDI score</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ForebrainDorsal_MixingUniformClusters_ToEstimateGDISensitivity.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Mixing Uniform Clusters To Estimate GDI Sensitivity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./GDI_Increment_From_Mixing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">GDI Increment From Mixing</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-9" aria-expanded="false">
 <span class="menu-text">Cluster evaluation</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-9" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-9" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./cd14_clusterizations_comparisons.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CD14+ clusterizations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./celltypist_annotation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CellTypist cell classification</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./clustering_comparison.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Clustering comparison</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ground_truth_comparison.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ground truth comparison</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-10" aria-expanded="false">
 <span class="menu-text">DEA evaluation</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-10" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-10" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./DEA_Type1Error.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Type I error</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./DEA_Seurat_COTAN.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Comparison Seurat - Cotan on Mouse Brain E15.0</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#preamble" id="toc-preamble" class="nav-link active" data-scroll-target="#preamble">Preamble</a></li>
  <li><a href="#loading-all-cotan-objects" id="toc-loading-all-cotan-objects" class="nav-link" data-scroll-target="#loading-all-cotan-objects">Loading all COTAN Objects</a></li>
  <li><a href="#relevant-clusters-lists" id="toc-relevant-clusters-lists" class="nav-link" data-scroll-target="#relevant-clusters-lists">Relevant clusters lists</a></li>
  <li><a href="#merge-clusters-from-different-datasets" id="toc-merge-clusters-from-different-datasets" class="nav-link" data-scroll-target="#merge-clusters-from-different-datasets">Merge clusters from different datasets</a></li>
  <li><a href="#create-all-raw-data-for-each-cluster-compatible-with-the-full-genes-list" id="toc-create-all-raw-data-for-each-cluster-compatible-with-the-full-genes-list" class="nav-link" data-scroll-target="#create-all-raw-data-for-each-cluster-compatible-with-the-full-genes-list">Create all raw data for each cluster compatible with the full genes’ list</a></li>
  <li><a href="#collect-size-and-gdi-for-all-selected-clusters-baseline-data" id="toc-collect-size-and-gdi-for-all-selected-clusters-baseline-data" class="nav-link" data-scroll-target="#collect-size-and-gdi-for-all-selected-clusters-baseline-data">Collect <code>size</code> and <code>GDI</code> for all selected clusters (baseline data)</a></li>
  <li><a href="#load-baseline-data" id="toc-load-baseline-data" class="nav-link" data-scroll-target="#load-baseline-data">Load baseline data</a></li>
  <li><a href="#calculate-the-gdi-of-the-mixtures-of-clusters" id="toc-calculate-the-gdi-of-the-mixtures-of-clusters" class="nav-link" data-scroll-target="#calculate-the-gdi-of-the-mixtures-of-clusters">Calculate the GDI of the mixtures of clusters</a>
  <ul class="collapse">
  <li><a href="#this-is-to-be-run-once-per-wanted-mixing-fraction" id="toc-this-is-to-be-run-once-per-wanted-mixing-fraction" class="nav-link" data-scroll-target="#this-is-to-be-run-once-per-wanted-mixing-fraction">This is to be run once per wanted <em>mixing-fraction</em></a></li>
  </ul></li>
  <li><a href="#load-calculated-data-for-analysis" id="toc-load-calculated-data-for-analysis" class="nav-link" data-scroll-target="#load-calculated-data-for-analysis">Load calculated data for analysis</a></li>
  <li><a href="#merge-all-results-and-calculate-the-fitting-regression-for-each-cluster-pair" id="toc-merge-all-results-and-calculate-the-fitting-regression-for-each-cluster-pair" class="nav-link" data-scroll-target="#merge-all-results-and-calculate-the-fitting-regression-for-each-cluster-pair">Merge all results and calculate the fitting regression for each cluster pair</a></li>
  <li><a href="#recall-cluster-distance-and-add-it-to-the-results" id="toc-recall-cluster-distance-and-add-it-to-the-results" class="nav-link" data-scroll-target="#recall-cluster-distance-and-add-it-to-the-results">Recall cluster distance and add it to the results</a></li>
  <li><a href="#merge-all-data-and-plot-it-using-a-priory-squared-distance-as-discriminant" id="toc-merge-all-data-and-plot-it-using-a-priory-squared-distance-as-discriminant" class="nav-link" data-scroll-target="#merge-all-data-and-plot-it-using-a-priory-squared-distance-as-discriminant">Merge all data and plot it using <em>a-priory</em> (squared) distance as discriminant</a></li>
  <li><a href="#calculate-the-gdi-of-the-mixtures-of-three-clusters" id="toc-calculate-the-gdi-of-the-mixtures-of-three-clusters" class="nav-link" data-scroll-target="#calculate-the-gdi-of-the-mixtures-of-three-clusters">Calculate the GDI of the mixtures of three clusters</a>
  <ul class="collapse">
  <li><a href="#the-total-mixture-is-20" id="toc-the-total-mixture-is-20" class="nav-link" data-scroll-target="#the-total-mixture-is-20">the total mixture is 20%</a></li>
  </ul></li>
  <li><a href="#estimate-predicted-gdi-increment-using-resultts-from-two-clusters-mixings" id="toc-estimate-predicted-gdi-increment-using-resultts-from-two-clusters-mixings" class="nav-link" data-scroll-target="#estimate-predicted-gdi-increment-using-resultts-from-two-clusters-mixings">Estimate predicted <code>GDI</code> increment using resultts from two-clusters mixings</a>
  <ul class="collapse">
  <li><a href="#the-estimates-assume-gdi-increment-proportional-to-mixture" id="toc-the-estimates-assume-gdi-increment-proportional-to-mixture" class="nav-link" data-scroll-target="#the-estimates-assume-gdi-increment-proportional-to-mixture">The estimates <em>assume</em> GDI increment proportional to mixture</a></li>
  </ul></li>
  <li><a href="#load-calculated-data-for-analysis-1" id="toc-load-calculated-data-for-analysis-1" class="nav-link" data-scroll-target="#load-calculated-data-for-analysis-1">Load calculated data for analysis</a></li>
  <li><a href="#compare-estimated-vs-real-gdi-increment" id="toc-compare-estimated-vs-real-gdi-increment" class="nav-link" data-scroll-target="#compare-estimated-vs-real-gdi-increment">Compare estimated vs real GDI increment</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Mixing Uniform Clusters To Estimate GDI Sensitivity</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Marco Fantozzi </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">January 18, 2024</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="preamble" class="level2">
<h2 class="anchored" data-anchor-id="preamble">Preamble</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(assertthat)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rlang)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(zeallot)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(COTAN)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(parallelDist)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">parallelly.fork.enable =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>inDir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"Data/MouseCortexFromLoom/"</span>)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>outDir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"Results/GDI_Sensitivity"</span>)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(outDir)) {</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(outDir)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="fu">setLoggingLevel</span>(<span class="dv">2</span>)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="fu">setLoggingFile</span>(<span class="fu">file.path</span>(outDir, <span class="st">"MixingClustersGDI_ForebrainDorsal.log"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="loading-all-cotan-objects" class="level2">
<h2 class="anchored" data-anchor-id="loading-all-cotan-objects">Loading all COTAN Objects</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>fb135Obj <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="fu">file.path</span>(inDir, <span class="st">"e13.5_ForebrainDorsal.cotan.RDS"</span>))</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">getMetadataElement</span>(fb135Obj, <span class="fu">datasetTags</span>()[[<span class="st">"cond"</span>]])</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>fb150Obj <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="fu">file.path</span>(inDir, <span class="st">"e15.0_ForebrainDorsal.cotan.RDS"</span>))</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">getMetadataElement</span>(fb150Obj, <span class="fu">datasetTags</span>()[[<span class="st">"cond"</span>]])</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>fb175Obj <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="fu">file.path</span>(inDir, <span class="st">"e17.5_ForebrainDorsal.cotan.RDS"</span>))</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="fu">getMetadataElement</span>(fb175Obj, <span class="fu">datasetTags</span>()[[<span class="st">"cond"</span>]])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="relevant-clusters-lists" class="level2">
<h2 class="anchored" data-anchor-id="relevant-clusters-lists">Relevant clusters lists</h2>
</section>
<section id="merge-clusters-from-different-datasets" class="level2">
<h2 class="anchored" data-anchor-id="merge-clusters-from-different-datasets">Merge clusters from different datasets</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>allGenes <span class="ot">&lt;-</span> <span class="fu">union</span>(<span class="fu">union</span>(<span class="fu">getGenes</span>(fb135Obj), <span class="fu">getGenes</span>(fb150Obj)), <span class="fu">getGenes</span>(fb175Obj))</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="co"># This function completes the raw data so that the genes list</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co"># matches the `allGenes` above</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>fillGenes <span class="ot">&lt;-</span> <span class="cf">function</span>(m, prefixName) {</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># find the missing genes in the given matrix m</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  missingGenes <span class="ot">&lt;-</span> <span class="sc">!</span>(allGenes <span class="sc">%in%</span> <span class="fu">rownames</span>(m))</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># create a matrix with the right sizes for the missing genes</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># and set their values all to zero</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>  fillM <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">0</span>, <span class="at">nrow =</span> <span class="fu">sum</span>(missingGenes), <span class="at">ncol =</span> <span class="fu">ncol</span>(m))</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(fillM) <span class="ot">&lt;-</span> allGenes[missingGenes]</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># append the new matrix to the original and reorder the rows</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># to match overall genes order</span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>  retM <span class="ot">&lt;-</span> <span class="fu">rbind</span>(m, fillM)</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>  retM <span class="ot">&lt;-</span> retM[allGenes, ]</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># assign unique identifiers to the columns, so that one can discriminate</span></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># the source even after merging with other matrices</span></span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(retM) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(prefixName, <span class="st">"_"</span>, <span class="fu">colnames</span>(retM))</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(retM)</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="create-all-raw-data-for-each-cluster-compatible-with-the-full-genes-list" class="level2">
<h2 class="anchored" data-anchor-id="create-all-raw-data-for-each-cluster-compatible-with-the-full-genes-list">Create all raw data for each cluster compatible with the full genes’ list</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>allClustersRawData <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (dsName <span class="cf">in</span> <span class="fu">names</span>(selected)) {</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  clList <span class="ot">&lt;-</span> <span class="fu">toClustersList</span>(<span class="fu">getClusters</span>(<span class="fu">objSelector</span>(dsName),</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">clName =</span> <span class="st">"original.clusters"</span>))</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (clName <span class="cf">in</span> selected[[dsName]]) {</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>    fullName <span class="ot">&lt;-</span> <span class="fu">paste0</span>(dsName, <span class="st">":"</span>, clName)</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>    fullRawData <span class="ot">&lt;-</span> <span class="fu">fillGenes</span>(<span class="fu">getRawData</span>(<span class="fu">objSelector</span>(dsName))[, clList[[clName]]],</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>                             fullName)</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>    allClustersRawData <span class="ot">&lt;-</span> <span class="fu">append</span>(allClustersRawData, <span class="fu">list</span>(fullRawData))</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(allClustersRawData)[<span class="fu">length</span>(allClustersRawData)] <span class="ot">&lt;-</span> fullName</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="fu">assert_that</span>(<span class="fu">identical</span>(<span class="fu">rownames</span>(allClustersRawData[[<span class="dv">1</span>]]), allGenes))</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a><span class="co"># delete the COTAN objects</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(fb135Obj, fb150Obj, fb175Obj)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="collect-size-and-gdi-for-all-selected-clusters-baseline-data" class="level2">
<h2 class="anchored" data-anchor-id="collect-size-and-gdi-for-all-selected-clusters-baseline-data">Collect <code>size</code> and <code>GDI</code> for all selected clusters (baseline data)</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>clDataFB135 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">file.path</span>(inDir, <span class="st">"e13.5_ForebrainDorsal_GDI_of_original_clusterization.RDS"</span>))</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>clDataFB150 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">file.path</span>(inDir, <span class="st">"e15.0_ForebrainDorsal_GDI_of_original_clusterization.RDS"</span>))</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>clDataFB175 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">file.path</span>(inDir, <span class="st">"e17.5_ForebrainDorsal_GDI_of_original_clusterization.RDS"</span>))</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>df1 <span class="ot">&lt;-</span> clDataFB135[selected[[<span class="st">"E13.5"</span>]], <span class="fu">c</span>(<span class="st">"size"</span>, <span class="st">"X1stPercentile"</span>), drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(df1) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"E13.5:"</span>, <span class="fu">rownames</span>(df1))</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>df2 <span class="ot">&lt;-</span> clDataFB150[selected[[<span class="st">"E15.0"</span>]], <span class="fu">c</span>(<span class="st">"size"</span>, <span class="st">"X1stPercentile"</span>), drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(df2) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"E15.0:"</span>, <span class="fu">rownames</span>(df2))</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>df3 <span class="ot">&lt;-</span> clDataFB175[selected[[<span class="st">"E17.5"</span>]], <span class="fu">c</span>(<span class="st">"size"</span>, <span class="st">"X1stPercentile"</span>), drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(df3) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"E17.5:"</span>, <span class="fu">rownames</span>(df3))</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>baselineGDI <span class="ot">&lt;-</span> <span class="fu">rbind</span>(df1, df2, df3)</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(baselineGDI) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"size"</span>, <span class="st">"GDI"</span>)</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(clDataFB135, clDataFB150, clDataFB175)</span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(df1, df2, df3)</span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(baselineGDI,</span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>        <span class="at">file =</span> <span class="fu">file.path</span>(outDir, <span class="st">"ForebrainDorsal_ClusterizationMixing_BaselineGDI.RDS"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-baseline-data" class="level2">
<h2 class="anchored" data-anchor-id="load-baseline-data">Load baseline data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>baselineGDI <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">file.path</span>(outDir, <span class="st">"ForebrainDorsal_ClusterizationMixing_BaselineGDI.RDS"</span>))</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co"># check good alignment of data</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="fu">assert_that</span>(<span class="fu">identical</span>(<span class="fu">rownames</span>(baselineGDI), <span class="fu">names</span>(allClustersRawData)))</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="fu">assert_that</span>(<span class="fu">identical</span>(baselineGDI[, <span class="st">"size"</span>],</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">vapply</span>(allClustersRawData, ncol,</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>                             <span class="at">FUN.VALUE =</span> <span class="fu">integer</span>(1L), <span class="at">USE.NAMES =</span> <span class="cn">FALSE</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="calculate-the-gdi-of-the-mixtures-of-clusters" class="level2">
<h2 class="anchored" data-anchor-id="calculate-the-gdi-of-the-mixtures-of-clusters">Calculate the GDI of the mixtures of clusters</h2>
<section id="this-is-to-be-run-once-per-wanted-mixing-fraction" class="level3">
<h3 class="anchored" data-anchor-id="this-is-to-be-run-once-per-wanted-mixing-fraction">This is to be run once per wanted <em>mixing-fraction</em></h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># small run</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">137</span>)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>mixingFraction <span class="ot">&lt;-</span> <span class="fl">0.80</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>mixingStr <span class="ot">&lt;-</span> <span class="fu">str_pad</span>(scales<span class="sc">::</span><span class="fu">label_percent</span>()(mixingFraction), <span class="dv">3</span>, <span class="at">pad =</span> <span class="st">"0"</span>)</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (mainName <span class="cf">in</span> <span class="fu">rownames</span>(baselineGDI)) {</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>  mainSize <span class="ot">&lt;-</span> baselineGDI[mainName, <span class="st">"size"</span>]</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>  mainGDI  <span class="ot">&lt;-</span> baselineGDI[mainName, <span class="st">"GDI"</span>]</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (clName <span class="cf">in</span> <span class="fu">rownames</span>(baselineGDI)) {</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (clName <span class="sc">==</span> mainName) <span class="cf">next</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">logThis</span>(<span class="fu">paste</span>(<span class="st">"Mixing"</span>, mainName, <span class="st">"with extra"</span>,</span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>                  mixingStr, <span class="st">"cells from"</span>, clName), <span class="at">logLevel =</span> <span class="dv">1</span>)</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>    clSize <span class="ot">&lt;-</span> baselineGDI[clName, <span class="st">"size"</span>]</span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>    actuallyMixedCells <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">ceiling</span>(mixingFraction <span class="sc">*</span> mainSize), clSize)</span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>    actualFraction <span class="ot">&lt;-</span> actuallyMixedCells <span class="sc">/</span> mainSize</span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>    sampleRawData <span class="ot">&lt;-</span> allClustersRawData[[clName]][, <span class="fu">sample</span>(clSize, actuallyMixedCells)]</span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>    mergedRawData <span class="ot">&lt;-</span> <span class="fu">cbind</span>(allClustersRawData[[mainName]], sampleRawData)</span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rm</span>(sampleRawData)</span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate the merged COEX</span></span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a>    cond <span class="ot">&lt;-</span> <span class="fu">paste0</span>(mainName, <span class="st">"|"</span>, mixingStr, <span class="st">"|"</span>, clName)</span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>    mergedObj <span class="ot">&lt;-</span> <span class="fu">automaticCOTANObjectCreation</span>(<span class="at">raw =</span> mergedRawData,</span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">GEO =</span> <span class="st">"MergedClusters"</span>,</span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">sequencingMethod =</span> <span class="st">"10X"</span>,</span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">sampleCondition =</span> cond,</span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">calcCoex =</span> <span class="cn">TRUE</span>, <span class="at">cores =</span> 10L,</span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">saveObj =</span> <span class="cn">FALSE</span>, <span class="at">outDir =</span> outDir)</span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rm</span>(mergedRawData)</span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Extract the GDI quantile</span></span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a>    mergedGDIData <span class="ot">&lt;-</span> <span class="fu">calculateGDI</span>(mergedObj)</span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rm</span>(mergedObj)</span>
<span id="cb7-40"><a href="#cb7-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-41"><a href="#cb7-41" aria-hidden="true" tabindex="-1"></a>    gdi <span class="ot">&lt;-</span> mergedGDIData[[<span class="st">"GDI"</span>]]</span>
<span id="cb7-42"><a href="#cb7-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(gdi) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(mergedGDIData)</span>
<span id="cb7-43"><a href="#cb7-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rm</span>(mergedGDIData)</span>
<span id="cb7-44"><a href="#cb7-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-45"><a href="#cb7-45" aria-hidden="true" tabindex="-1"></a>    gdi <span class="ot">&lt;-</span> <span class="fu">sort</span>(gdi, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-46"><a href="#cb7-46" aria-hidden="true" tabindex="-1"></a>    lastPercentile <span class="ot">&lt;-</span> <span class="fu">quantile</span>(gdi, <span class="at">probs =</span> <span class="fl">0.99</span>)</span>
<span id="cb7-47"><a href="#cb7-47" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rm</span>(gdi)</span>
<span id="cb7-48"><a href="#cb7-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-49"><a href="#cb7-49" aria-hidden="true" tabindex="-1"></a>    results <span class="ot">&lt;-</span> <span class="fu">rbind</span>(results,</span>
<span id="cb7-50"><a href="#cb7-50" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">data.frame</span>(<span class="st">"MainCluster"</span> <span class="ot">=</span> mainName, <span class="st">"OtherCluster"</span> <span class="ot">=</span> clName,</span>
<span id="cb7-51"><a href="#cb7-51" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"MixingFraction"</span> <span class="ot">=</span> actualFraction, <span class="st">"GDI"</span> <span class="ot">=</span> lastPercentile,</span>
<span id="cb7-52"><a href="#cb7-52" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"GDIIncrement"</span> <span class="ot">=</span> lastPercentile <span class="sc">-</span> mainGDI))</span>
<span id="cb7-53"><a href="#cb7-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-54"><a href="#cb7-54" aria-hidden="true" tabindex="-1"></a>    <span class="fu">logThis</span>(<span class="fu">paste</span>(<span class="st">"Mixing"</span>, mainName, <span class="st">"with"</span>, clName,</span>
<span id="cb7-55"><a href="#cb7-55" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"accomplished with GDI"</span>, lastPercentile), <span class="at">logLevel =</span> <span class="dv">1</span>)</span>
<span id="cb7-56"><a href="#cb7-56" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb7-57"><a href="#cb7-57" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-58"><a href="#cb7-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-59"><a href="#cb7-59" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(results) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb7-60"><a href="#cb7-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-61"><a href="#cb7-61" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(results, <span class="fu">file.path</span>(outDir, <span class="fu">paste0</span>(<span class="st">"GDI_with_"</span>, mixingStr, <span class="st">"_Mixing.RDS"</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="load-calculated-data-for-analysis" class="level2">
<h2 class="anchored" data-anchor-id="load-calculated-data-for-analysis">Load calculated data for analysis</h2>
</section>
<section id="merge-all-results-and-calculate-the-fitting-regression-for-each-cluster-pair" class="level2">
<h2 class="anchored" data-anchor-id="merge-all-results-and-calculate-the-fitting-regression-for-each-cluster-pair">Merge all results and calculate the fitting regression for each cluster pair</h2>
</section>
<section id="recall-cluster-distance-and-add-it-to-the-results" class="level2">
<h2 class="anchored" data-anchor-id="recall-cluster-distance-and-add-it-to-the-results">Recall cluster distance and add it to the results</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co">#zeroOneAvg &lt;- readRDS(file.path(inDir, "ClustersDistances", "allZeroOne.RDS"))</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>zeroOneAvg <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">file.path</span>(outDir, <span class="st">"distanceZeroOne.RDS"</span>))</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>distZeroOne <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">parDist</span>(<span class="fu">t</span>(zeroOneAvg), <span class="at">method =</span> <span class="st">"hellinger"</span>, <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>))<span class="sc">^</span><span class="dv">2</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>distZeroOneLong <span class="ot">&lt;-</span> <span class="fu">rownames_to_column</span>(<span class="fu">as.data.frame</span>(distZeroOne), <span class="at">var =</span> <span class="st">"MainCluster"</span>)</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>distZeroOneLong <span class="ot">&lt;-</span><span class="fu">pivot_longer</span>(distZeroOneLong,</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>                               <span class="at">cols =</span> <span class="sc">!</span>MainCluster,</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>                               <span class="at">names_to =</span> <span class="st">"OtherCluster"</span>, </span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>                               <span class="at">values_to =</span> <span class="st">"Distance"</span>)</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>distZeroOneLong <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(distZeroOneLong[distZeroOneLong[[<span class="st">"Distance"</span>]] <span class="sc">!=</span> <span class="fl">0.0</span>, ])</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a><span class="co">#assert_that(identical(distZeroOneLong[, 1:2], resMix20[, 1:2]))</span></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>perm <span class="ot">&lt;-</span> <span class="fu">order</span>(distZeroOneLong[[<span class="st">"Distance"</span>]])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Scatter plot of the effective increment [Y] against estimated increment [X]</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>distDF <span class="ot">&lt;-</span> <span class="fu">cbind</span>(distZeroOneLong[, <span class="st">"Distance"</span>, <span class="at">drop =</span> <span class="cn">FALSE</span>],</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>                <span class="fu">sqrt</span>(distZeroOneLong[, <span class="st">"Distance"</span>, <span class="at">drop =</span> <span class="cn">FALSE</span>]))</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(distDF) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Distance"</span>, <span class="st">"DistanceSqrt"</span>)</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>D2IPlot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">cbind</span>(resMix40, distDF),</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">aes</span>(<span class="at">x=</span>Distance, <span class="at">y=</span>GDIIncrement)) <span class="sc">+</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>             <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>             <span class="fu">geom_smooth</span>(<span class="at">method=</span>lm, <span class="at">formula =</span> y <span class="sc">~</span> x) </span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>           <span class="co"># +  xlim(0, 1.5) + ylim(0, 1.5) + coord_fixed()</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(D2IPlot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="merge-all-data-and-plot-it-using-a-priory-squared-distance-as-discriminant" class="level2">
<h2 class="anchored" data-anchor-id="merge-all-data-and-plot-it-using-a-priory-squared-distance-as-discriminant">Merge all data and plot it using <em>a-priory</em> (squared) distance as discriminant</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>allRes <span class="ot">&lt;-</span> <span class="fu">rbind</span>(resMix05[perm, ], resMix10[perm, ], resMix20[perm, ], resMix40[perm, ], resMix80[perm, ])</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(allRes) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>allRes <span class="ot">&lt;-</span> <span class="fu">cbind</span>(allRes, <span class="st">"ClusterPair"</span> <span class="ot">=</span> <span class="fu">rep.int</span>(<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">210</span>),<span class="dv">5</span>))</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>allRes <span class="ot">&lt;-</span> <span class="fu">cbind</span>(allRes, <span class="st">"Distance"</span> <span class="ot">=</span> <span class="fu">rep</span>(distZeroOneLong[[<span class="st">"Distance"</span>]][perm], <span class="dv">5</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>mg <span class="ot">&lt;-</span> <span class="cf">function</span>(mixing) { <span class="fu">ceiling</span>(<span class="fu">log2</span>(<span class="fu">round</span>(mixing<span class="sc">*</span><span class="dv">40</span>))) }</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>IScPlot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(allRes, <span class="fu">aes</span>(<span class="at">x=</span><span class="fu">mg</span>(MixingFraction), <span class="at">y=</span>GDIIncrement, <span class="at">color=</span>Distance,</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">group =</span> <span class="fu">mg</span>(MixingFraction))) <span class="sc">+</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_point() +</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="co">#  scale_color_continuous(type = "viridis") +</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="co"># geom_point(aes(size=log10(GDIIncrement+10))) +</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="co">#  scale_x_log10()</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(IScPlot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="ForebrainDorsal_MixingUniformClusters_ToEstimateGDISensitivity_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>perm2 <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">210</span>, <span class="at">each =</span> <span class="dv">5</span>) <span class="sc">+</span> <span class="fu">rep</span>(<span class="fu">seq</span>(<span class="dv">1</span>, <span class="dv">5</span> <span class="sc">*</span> <span class="dv">210</span>, <span class="at">by =</span> <span class="dv">210</span>), <span class="at">times =</span> <span class="dv">5</span>) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>allRes2 <span class="ot">&lt;-</span> allRes[perm2, ]</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="co"># c(1,44) - c(84,127) - c(167,210)</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>ILinesPLot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(allRes2[allRes2[[<span class="st">"ClusterPair"</span>]] <span class="sc">%between%</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">210</span>), ],</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">aes</span>(<span class="at">x =</span> MixingFraction, <span class="at">y =</span> GDIIncrement,</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">color =</span> (ClusterPair <span class="sc">-</span> <span class="dv">1</span>) <span class="sc">%/%</span> <span class="dv">42</span> <span class="sc">+</span> <span class="fl">0.5</span>)) <span class="sc">+</span> </span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_path</span>(<span class="fu">aes</span>(<span class="at">group =</span> ClusterPair)) <span class="sc">+</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>  <span class="co">#scale_x_log10() + </span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_stepsn</span>(<span class="at">colours =</span> <span class="fu">hcl.colors</span>(<span class="dv">5</span>, <span class="at">palette =</span> <span class="st">"Dark 2"</span>)[<span class="dv">5</span><span class="sc">:</span><span class="dv">1</span>])</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(ILinesPLot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="ForebrainDorsal_MixingUniformClusters_ToEstimateGDISensitivity_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="calculate-the-gdi-of-the-mixtures-of-three-clusters" class="level2">
<h2 class="anchored" data-anchor-id="calculate-the-gdi-of-the-mixtures-of-three-clusters">Calculate the GDI of the mixtures of three clusters</h2>
<section id="the-total-mixture-is-20" class="level3">
<h3 class="anchored" data-anchor-id="the-total-mixture-is-20">the total mixture is 20%</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># small run</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">137</span>)</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>numPairsPerCluster <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>mixingFraction <span class="ot">&lt;-</span> <span class="fl">0.20</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (mainName <span class="cf">in</span> <span class="fu">rownames</span>(baselineGDI)) {</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>  mainSize <span class="ot">&lt;-</span> baselineGDI[mainName, <span class="st">"size"</span>]</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>  mainGDI  <span class="ot">&lt;-</span> baselineGDI[mainName, <span class="st">"GDI"</span>]</span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>  executedPairs <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span>numPairsPerCluster)) {</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>    <span class="cf">repeat</span> {</span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>      clPair <span class="ot">&lt;-</span> <span class="fu">rownames</span>(baselineGDI)[<span class="fu">sample</span>(<span class="fu">nrow</span>(baselineGDI), <span class="dv">2</span>)]</span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>      clPair <span class="ot">&lt;-</span> clPair[<span class="fu">order</span>(clPair)]</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a>      clPairName <span class="ot">&lt;-</span> <span class="fu">paste0</span>(clPair[[1L]], <span class="st">":"</span>, clPair[[2L]])</span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (<span class="sc">!</span>(mainName <span class="sc">%in%</span> clPair) <span class="sc">&amp;&amp;</span> <span class="sc">!</span>(clPairName <span class="sc">%in%</span> executedPairs)) <span class="cf">break</span></span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">logThis</span>(<span class="fu">paste</span>(<span class="st">"Mixing"</span>, mainName, <span class="st">"with extra 20% cells from"</span>,</span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a>                  clPair[[1L]], <span class="st">"and"</span>, clPair[[2L]]), <span class="at">logLevel =</span> <span class="dv">1</span>)</span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a>    mixingFraction1 <span class="ot">&lt;-</span> <span class="fu">runif</span>(1L, <span class="fl">0.25</span> <span class="sc">*</span> mixingFraction, <span class="fl">0.75</span> <span class="sc">*</span> mixingFraction)</span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a>    mixingFraction2 <span class="ot">&lt;-</span> mixingFraction <span class="sc">-</span> mixingFraction1</span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a>    clSize1 <span class="ot">&lt;-</span> baselineGDI[clPair[[1L]], <span class="st">"size"</span>]</span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a>    clSize2 <span class="ot">&lt;-</span> baselineGDI[clPair[[2L]], <span class="st">"size"</span>]</span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a>    actuallyMixedCells1 <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">ceiling</span>(mixingFraction1 <span class="sc">*</span> mainSize), clSize1)</span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a>    actuallyMixedCells2 <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">ceiling</span>(mixingFraction2 <span class="sc">*</span> mainSize), clSize2)</span>
<span id="cb13-35"><a href="#cb13-35" aria-hidden="true" tabindex="-1"></a>    actualFraction1 <span class="ot">&lt;-</span> actuallyMixedCells1 <span class="sc">/</span> mainSize</span>
<span id="cb13-36"><a href="#cb13-36" aria-hidden="true" tabindex="-1"></a>    actualFraction2 <span class="ot">&lt;-</span> actuallyMixedCells2 <span class="sc">/</span> mainSize</span>
<span id="cb13-37"><a href="#cb13-37" aria-hidden="true" tabindex="-1"></a>    sampleRawData1 <span class="ot">&lt;-</span> allClustersRawData[[clPair[[1L]]]][, <span class="fu">sample</span>(clSize1, actuallyMixedCells1)]</span>
<span id="cb13-38"><a href="#cb13-38" aria-hidden="true" tabindex="-1"></a>    sampleRawData2 <span class="ot">&lt;-</span> allClustersRawData[[clPair[[2L]]]][, <span class="fu">sample</span>(clSize2, actuallyMixedCells2)]</span>
<span id="cb13-39"><a href="#cb13-39" aria-hidden="true" tabindex="-1"></a>    mergedRawData <span class="ot">&lt;-</span> <span class="fu">cbind</span>(allClustersRawData[[mainName]], sampleRawData1, sampleRawData2)</span>
<span id="cb13-40"><a href="#cb13-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rm</span>(sampleRawData1, sampleRawData2)</span>
<span id="cb13-41"><a href="#cb13-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-42"><a href="#cb13-42" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate the merged COEX</span></span>
<span id="cb13-43"><a href="#cb13-43" aria-hidden="true" tabindex="-1"></a>    cond <span class="ot">&lt;-</span> <span class="fu">paste0</span>(mainName, <span class="st">"|20%|"</span>, clPairName)</span>
<span id="cb13-44"><a href="#cb13-44" aria-hidden="true" tabindex="-1"></a>    mergedObj <span class="ot">&lt;-</span> <span class="fu">automaticCOTANObjectCreation</span>(<span class="at">raw =</span> mergedRawData,</span>
<span id="cb13-45"><a href="#cb13-45" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">GEO =</span> <span class="st">"MergedClusters"</span>,</span>
<span id="cb13-46"><a href="#cb13-46" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">sequencingMethod =</span> <span class="st">"10X"</span>,</span>
<span id="cb13-47"><a href="#cb13-47" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">sampleCondition =</span> cond,</span>
<span id="cb13-48"><a href="#cb13-48" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">calcCoex =</span> <span class="cn">TRUE</span>, <span class="at">cores =</span> 10L,</span>
<span id="cb13-49"><a href="#cb13-49" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">saveObj =</span> <span class="cn">FALSE</span>, <span class="at">outDir =</span> outDir)</span>
<span id="cb13-50"><a href="#cb13-50" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rm</span>(mergedRawData)</span>
<span id="cb13-51"><a href="#cb13-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-52"><a href="#cb13-52" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Extract the GDI quantile</span></span>
<span id="cb13-53"><a href="#cb13-53" aria-hidden="true" tabindex="-1"></a>    mergedGDIData <span class="ot">&lt;-</span> <span class="fu">calculateGDI</span>(mergedObj)</span>
<span id="cb13-54"><a href="#cb13-54" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rm</span>(mergedObj)</span>
<span id="cb13-55"><a href="#cb13-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-56"><a href="#cb13-56" aria-hidden="true" tabindex="-1"></a>    gdi <span class="ot">&lt;-</span> mergedGDIData[[<span class="st">"GDI"</span>]]</span>
<span id="cb13-57"><a href="#cb13-57" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(gdi) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(mergedGDIData)</span>
<span id="cb13-58"><a href="#cb13-58" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rm</span>(mergedGDIData)</span>
<span id="cb13-59"><a href="#cb13-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-60"><a href="#cb13-60" aria-hidden="true" tabindex="-1"></a>    gdi <span class="ot">&lt;-</span> <span class="fu">sort</span>(gdi, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span>
<span id="cb13-61"><a href="#cb13-61" aria-hidden="true" tabindex="-1"></a>    lastPercentile <span class="ot">&lt;-</span> <span class="fu">quantile</span>(gdi, <span class="at">probs =</span> <span class="fl">0.99</span>)</span>
<span id="cb13-62"><a href="#cb13-62" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rm</span>(gdi)</span>
<span id="cb13-63"><a href="#cb13-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-64"><a href="#cb13-64" aria-hidden="true" tabindex="-1"></a>    results <span class="ot">&lt;-</span> <span class="fu">rbind</span>(results,</span>
<span id="cb13-65"><a href="#cb13-65" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">data.frame</span>(<span class="st">"MainCluster"</span> <span class="ot">=</span> mainName,</span>
<span id="cb13-66"><a href="#cb13-66" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"OtherCluster1"</span> <span class="ot">=</span> clPair[[1L]],</span>
<span id="cb13-67"><a href="#cb13-67" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"MixingFraction1"</span> <span class="ot">=</span> actualFraction1,</span>
<span id="cb13-68"><a href="#cb13-68" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"OtherCluster2"</span> <span class="ot">=</span> clPair[[2L]],</span>
<span id="cb13-69"><a href="#cb13-69" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"MixingFraction2"</span> <span class="ot">=</span> actualFraction2,</span>
<span id="cb13-70"><a href="#cb13-70" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"GDI"</span> <span class="ot">=</span> lastPercentile,</span>
<span id="cb13-71"><a href="#cb13-71" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"GDIIncrement"</span> <span class="ot">=</span> lastPercentile <span class="sc">-</span> mainGDI))</span>
<span id="cb13-72"><a href="#cb13-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-73"><a href="#cb13-73" aria-hidden="true" tabindex="-1"></a>    <span class="fu">logThis</span>(<span class="fu">paste</span>(<span class="st">"Mixing"</span>, mainName, <span class="st">"with"</span>, clPairName,</span>
<span id="cb13-74"><a href="#cb13-74" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"accomplished with GDI"</span>, lastPercentile), <span class="at">logLevel =</span> <span class="dv">1</span>)</span>
<span id="cb13-75"><a href="#cb13-75" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb13-76"><a href="#cb13-76" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb13-77"><a href="#cb13-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-78"><a href="#cb13-78" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(results) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb13-79"><a href="#cb13-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-80"><a href="#cb13-80" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(results, <span class="fu">file.path</span>(outDir, <span class="fu">paste0</span>(<span class="st">"GDI_with_20%_Mixing_from_Pairs.RDS"</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="estimate-predicted-gdi-increment-using-resultts-from-two-clusters-mixings" class="level2">
<h2 class="anchored" data-anchor-id="estimate-predicted-gdi-increment-using-resultts-from-two-clusters-mixings">Estimate predicted <code>GDI</code> increment using resultts from two-clusters mixings</h2>
<section id="the-estimates-assume-gdi-increment-proportional-to-mixture" class="level3">
<h3 class="anchored" data-anchor-id="the-estimates-assume-gdi-increment-proportional-to-mixture">The estimates <em>assume</em> GDI increment proportional to mixture</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>resMix20_2 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">file.path</span>(outDir, <span class="fu">paste0</span>(<span class="st">"GDI_with_20%_Mixing_from_Pairs.RDS"</span>)))</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>resMix20_2 <span class="ot">&lt;-</span> <span class="fu">cbind</span>(resMix20_2, <span class="cn">NA</span>)</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(resMix20_2)[<span class="fu">ncol</span>(resMix20_2)] <span class="ot">&lt;-</span> <span class="st">"PredictedGDIIncrement"</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (r <span class="cf">in</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(resMix20_2))) {</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>  mainName <span class="ot">&lt;-</span> resMix20_2[r, <span class="st">"MainCluster"</span>]</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>  mainMatch20 <span class="ot">&lt;-</span> resMix20[resMix20[, <span class="st">"MainCluster"</span>] <span class="sc">==</span> mainName, ]</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>  pos1 <span class="ot">&lt;-</span> <span class="fu">which</span>(mainMatch20[, <span class="st">"OtherCluster"</span>] <span class="sc">==</span> resMix20_2[r, <span class="st">"OtherCluster1"</span>])</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>  pos2 <span class="ot">&lt;-</span> <span class="fu">which</span>(mainMatch20[, <span class="st">"OtherCluster"</span>] <span class="sc">==</span> resMix20_2[r, <span class="st">"OtherCluster2"</span>])</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>  GDIIncr1 <span class="ot">&lt;-</span> mainMatch20[pos1, <span class="st">"GDIIncrement"</span>]</span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>  GDIIncr2 <span class="ot">&lt;-</span> mainMatch20[pos2, <span class="st">"GDIIncrement"</span>]</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>  w1 <span class="ot">&lt;-</span> resMix20_2[r, <span class="st">"MixingFraction1"</span>] <span class="sc">/</span> mainMatch20[pos1, <span class="st">"MixingFraction"</span>]</span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>  w2 <span class="ot">&lt;-</span> resMix20_2[r, <span class="st">"MixingFraction2"</span>] <span class="sc">/</span> mainMatch20[pos2, <span class="st">"MixingFraction"</span>]</span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>  predictedGDIIncr <span class="ot">&lt;-</span> (w1 <span class="sc">*</span> GDIIncr1 <span class="sc">+</span> w2 <span class="sc">*</span> GDIIncr2)</span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>  resMix20_2[r, <span class="st">"PredictedGDIIncrement"</span>] <span class="ot">&lt;-</span> predictedGDIIncr</span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(resMix20_2, <span class="fu">file.path</span>(outDir, <span class="fu">paste0</span>(<span class="st">"GDI_with_20%_Mixing_from_Pairs.RDS"</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="load-calculated-data-for-analysis-1" class="level2">
<h2 class="anchored" data-anchor-id="load-calculated-data-for-analysis-1">Load calculated data for analysis</h2>
</section>
<section id="compare-estimated-vs-real-gdi-increment" class="level2">
<h2 class="anchored" data-anchor-id="compare-estimated-vs-real-gdi-increment">Compare estimated vs real GDI increment</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Scatter plot of the effective increment [Y] against estimated increment [X]</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>pg <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(resMix20_2, <span class="fu">aes</span>(<span class="at">x=</span>PredictedGDIIncrement, <span class="at">y=</span>GDIIncrement)) <span class="sc">+</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method=</span>lm, <span class="at">formula =</span> y <span class="sc">~</span> x <span class="sc">+</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_fixed</span>() <span class="sc">+</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>, <span class="fl">1.5</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="dv">0</span>, <span class="fl">1.5</span>)</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>  <span class="co">#scale_x_log10() + scale_y_log10() </span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(pg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="ForebrainDorsal_MixingUniformClusters_ToEstimateGDISensitivity_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>The plot shows that having the 20% extraneous cells in the mixture coming from multiple clusters does not affect significantly the sensitivity of the GDI to score cluster uniformity.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.3.2 (2023-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0 
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       
 [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   
 [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          
[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   

time zone: Europe/Rome
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] tidyr_1.3.0        tibble_3.2.1       parallelDist_0.2.6 COTAN_2.3.0       
 [5] data.table_1.14.8  zeallot_0.1.0      ggplot2_3.4.2      scales_1.3.0      
 [9] rlang_1.1.1        assertthat_0.2.1  

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.21          splines_4.3.2            
  [3] later_1.3.1               polyclip_1.10-4          
  [5] fastDummies_1.7.3         lifecycle_1.0.3          
  [7] doParallel_1.0.17         globals_0.16.2           
  [9] lattice_0.22-5            MASS_7.3-60              
 [11] dendextend_1.17.1         magrittr_2.0.3           
 [13] plotly_4.10.2             rmarkdown_2.24           
 [15] yaml_2.3.7                httpuv_1.6.11            
 [17] Seurat_5.0.0              sctransform_0.4.1        
 [19] spam_2.10-0               askpass_1.2.0            
 [21] sp_2.1-1                  spatstat.sparse_3.0-2    
 [23] reticulate_1.34.0         cowplot_1.1.1            
 [25] pbapply_1.7-2             RColorBrewer_1.1-3       
 [27] abind_1.4-5               Rtsne_0.16               
 [29] purrr_1.0.1               BiocGenerics_0.46.0      
 [31] circlize_0.4.15           IRanges_2.34.1           
 [33] S4Vectors_0.38.1          ggrepel_0.9.3            
 [35] irlba_2.3.5.1             listenv_0.9.0            
 [37] spatstat.utils_3.0-3      umap_0.2.10.0            
 [39] goftest_1.2-3             RSpectra_0.16-1          
 [41] spatstat.random_3.2-1     dqrng_0.3.0              
 [43] fitdistrplus_1.1-11       parallelly_1.36.0        
 [45] DelayedMatrixStats_1.22.5 leiden_0.4.3             
 [47] codetools_0.2-19          DelayedArray_0.26.7      
 [49] tidyselect_1.2.0          shape_1.4.6              
 [51] farver_2.1.1              ScaledMatrix_1.8.1       
 [53] viridis_0.6.4             matrixStats_1.2.0        
 [55] stats4_4.3.2              spatstat.explore_3.2-1   
 [57] jsonlite_1.8.7            GetoptLong_1.0.5         
 [59] ellipsis_0.3.2            progressr_0.14.0         
 [61] ggridges_0.5.4            survival_3.5-7           
 [63] iterators_1.0.14          foreach_1.5.2            
 [65] tools_4.3.2               ica_1.0-3                
 [67] Rcpp_1.0.11               glue_1.6.2               
 [69] gridExtra_2.3             xfun_0.39                
 [71] mgcv_1.9-1                MatrixGenerics_1.12.3    
 [73] ggthemes_5.0.0            dplyr_1.1.2              
 [75] withr_2.5.0               fastmap_1.1.1            
 [77] fansi_1.0.4               openssl_2.1.0            
 [79] digest_0.6.33             rsvd_1.0.5               
 [81] R6_2.5.1                  mime_0.12                
 [83] colorspace_2.1-0          scattermore_1.2          
 [85] tensor_1.5                spatstat.data_3.0-1      
 [87] utf8_1.2.3                generics_0.1.3           
 [89] httr_1.4.6                htmlwidgets_1.6.2        
 [91] S4Arrays_1.2.0            uwot_0.1.16              
 [93] pkgconfig_2.0.3           gtable_0.3.3             
 [95] ComplexHeatmap_2.16.0     lmtest_0.9-40            
 [97] htmltools_0.5.7           dotCall64_1.1-0          
 [99] clue_0.3-64               SeuratObject_5.0.0       
[101] png_0.1-8                 knitr_1.43               
[103] rstudioapi_0.15.0         reshape2_1.4.4           
[105] rjson_0.2.21              nlme_3.1-163             
[107] zoo_1.8-12                GlobalOptions_0.1.2      
[109] stringr_1.5.0             KernSmooth_2.23-22       
[111] parallel_4.3.2            miniUI_0.1.1.1           
[113] RcppZiggurat_0.1.6        pillar_1.9.0             
[115] grid_4.3.2                vctrs_0.6.3              
[117] RANN_2.6.1                promises_1.2.0.1         
[119] BiocSingular_1.16.0       beachmat_2.16.0          
[121] xtable_1.8-4              cluster_2.1.6            
[123] evaluate_0.21             cli_3.6.1                
[125] compiler_4.3.2            crayon_1.5.2             
[127] future.apply_1.11.0       labeling_0.4.2           
[129] plyr_1.8.8                stringi_1.8.1            
[131] viridisLite_0.4.2         deldir_2.0-2             
[133] BiocParallel_1.34.2       munsell_0.5.0            
[135] lazyeval_0.2.2            spatstat.geom_3.2-4      
[137] PCAtools_2.14.0           Matrix_1.6-3             
[139] RcppHNSW_0.5.0            patchwork_1.1.2          
[141] sparseMatrixStats_1.12.2  future_1.33.0            
[143] shiny_1.8.0               ROCR_1.0-11              
[145] Rfast_2.1.0               igraph_1.6.0             
[147] RcppParallel_5.1.7       </code></pre>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>