<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.47">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Silvia Galfrè">
<meta name="dcterms.date" content="2025-12-16">

<title>Real UT clusters – COTAN Datasets analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./NonCT_clusters.html">Cluster evaluation</a></li><li class="breadcrumb-item"><a href="./Real_UT_clustersByCotan.html">Real UT clusters with LR</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">COTAN Datasets analysis</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Home</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false">
 <span class="menu-text">Datasets</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false">
 <span class="menu-text">Mouse Cortex Loo 2019</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Mouse_Cortex_E14.5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cleaning and cell clustering</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneCoexAnalysisDataset1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Coexpression analaysis</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false">
 <span class="menu-text">Mouse Cortex Yuzwa 2017</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./CorticalCells_GSM2861511_E135-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">E13.5 Cleaning and cell clustering</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneCoexAnalysisE13.5Yuzwa.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Correlation analysis for E13.5</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./CorticalCells_GSM2861514_E175-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">E17.5 Cleaning and cell clustering</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneCoexAnalysisE17.5Yuzwa.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Correlation analysis for E17.5</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false">
 <span class="menu-text">Mouse Brain LaManno 2021</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ForebrainDorsal_E135-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">E13.5 Cleaning and cell clustering</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneCoexAnalysisE13.5MouseBrain.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Correlation analysis for E13.5</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ForebrainDorsal_E150-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">E15.0 Cleaning and cell clustering</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneCoexAnalysisE15.0MouseBrain.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Correlation analysis for E15.0</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ForebrainDorsal_E175-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">E17.5 Cleaning and cell clustering</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneCoexAnalysisE17.5MouseBrain.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Correlation analysis for E17.5</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ForebrainLaManno2021ClustersGDI.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Original Clusters GDI evaluation</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false">
 <span class="menu-text">CD14+</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./cd14_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CD14+ Cleaning and cell clustering</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" role="navigation" aria-expanded="false">
 <span class="menu-text">PBMC form 10X site</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./PBMC1_filtering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cleaning PBMC1</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./PBMC2_filtering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cleaning PBMC2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./PBMC3_filtering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cleaning PBMC3</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./PBMC4_filtering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cleaning PBMC4</span></a>
  </div>
</li>
      </ul>
  </li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" role="navigation" aria-expanded="false">
 <span class="menu-text">Coex score evaluation</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-7" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./generalCoexAssess.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">General Coex Evaluation</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" role="navigation" aria-expanded="false">
 <span class="menu-text">GDI sensitivity</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-8" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./OriginalClustersDistanceEvaluation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Clusters Distance Evaluation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./FindUniformOrigCluster.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cluster Unifomity and GDI score</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./GDI_Increment_From_Mixing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">GDI Increment From Mixing</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-9" role="navigation" aria-expanded="true">
 <span class="menu-text">Cluster evaluation</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-9" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-9" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./NonCT_clusters.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Not UT cluster with LR</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Real_UT_clustersByCotan.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Real UT clusters with LR</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./cd14_clusterizations_comparisons.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CD14+ clusterizations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./clustering_comparison.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PBMCs clustering comparison</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./probability_score.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PBMCs probability score</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-10" role="navigation" aria-expanded="false">
 <span class="menu-text">DEA evaluation</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-10" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-10" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./DEA_Type1Er.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Type I error</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneExpressionAnalysisUniformClusters.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Explorative analysis of gene expression</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./FDR_analisys.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">FDR analysis - define DE genes</span></a>
  </div>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-11" role="navigation" aria-expanded="false">
 <span class="menu-text">FDR and TPR - ROC curves</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-11" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-11" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./FDR_analisysResults3_10.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Results - thresholds true 3% 10%</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./FDR_analisysResults5_20.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Results - thresholds true 5% 20%</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./FDR_analisysResults5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Results - thresholds true 5%</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./FDR_analysisResultsLFC.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Results - Log Fold Change</span></a>
  </div>
</li>
      </ul>
  </li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-12" role="navigation" aria-expanded="false">
 <span class="menu-text">Additional code and scripts</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-12" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-12" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./src/scanpyTypeIError.py" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ScamPy Type I Error code</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./src/scanpyFDR.py" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ScamPy FDR code</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scanpy.ipynb" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Example of scanpy clustering and DEA</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scvi-tools.ipynb" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Example of ScVI clustering and DEA</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#preamble" id="toc-preamble" class="nav-link active" data-scroll-target="#preamble">Preamble</a></li>
  <li><a href="#and-434-e15.0" id="toc-and-434-e15.0" class="nav-link" data-scroll-target="#and-434-e15.0">428 and 434 e15.0</a></li>
  <li><a href="#e13.5-e14.5" id="toc-e13.5-e14.5" class="nav-link" data-scroll-target="#e13.5-e14.5">434 e13.5 e14.5</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./NonCT_clusters.html">Cluster evaluation</a></li><li class="breadcrumb-item"><a href="./Real_UT_clustersByCotan.html">Real UT clusters with LR</a></li></ol></nav>
<div class="quarto-title">
<h1 class="title">Real UT clusters</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Silvia Galfrè </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">December 16, 2025</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<p>Cl432 E13.5 E15.0</p>
<section id="preamble" class="level2">
<h2 class="anchored" data-anchor-id="preamble">Preamble</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(zeallot)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(parallelDist)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(caret)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_bw</span>())</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(COTAN)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(nnet)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">parallelly.fork.enable =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>inDir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"Data/MouseCortexFromLoom/"</span>)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="co">#setLoggingLevel(2)</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="co">#setLoggingFile(file.path(inDir, "MixingClustersGDI_ForebrainDorsal.log"))</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>outDir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"Data/MouseCortexFromLoom/PureClusters/"</span>)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(outDir)) {</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(outDir)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="and-434-e15.0" class="level2">
<h2 class="anchored" data-anchor-id="and-434-e15.0">428 and 434 e15.0</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>e15<span class="fl">.0</span> <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">paste0</span>(inDir,<span class="st">"e15.0_ForebrainDorsal.cotan.RDS"</span>))</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>cells <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rownames</span>(<span class="fu">getMetadataCells</span>(e15<span class="fl">.0</span>)[<span class="fu">getMetadataCells</span>(e15<span class="fl">.0</span>)[,<span class="st">"Clusters"</span>] <span class="sc">==</span> <span class="dv">428</span>,]),</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>           <span class="fu">rownames</span>(<span class="fu">getMetadataCells</span>(e15<span class="fl">.0</span>)[<span class="fu">getMetadataCells</span>(e15<span class="fl">.0</span>)[,<span class="st">"Clusters"</span>] <span class="sc">==</span> <span class="dv">434</span>,]))</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>cl428_cl434 <span class="ot">&lt;-</span> <span class="fu">dropGenesCells</span>(e15<span class="fl">.0</span>,<span class="at">cells =</span> <span class="fu">getCells</span>(e15<span class="fl">.0</span>)[<span class="sc">!</span><span class="fu">getCells</span>(e15<span class="fl">.0</span>) <span class="sc">%in%</span> cells] )</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>cl428_cl434 <span class="ot">&lt;-</span> <span class="fu">clean</span>(cl428_cl434)</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>cl428_cl434 <span class="ot">&lt;-</span> <span class="fu">proceedToCoex</span>(cl428_cl434)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="co">#saveRDS(cl428_cl434,past0(outDir,"cl428_cl434.cotan.RDS"))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>cl428_cl434GDI <span class="ot">&lt;-</span> <span class="fu">calculateGDI</span>(cl428_cl434)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>subsetGDIcl428_cl434 <span class="ot">&lt;-</span> cl428_cl434GDI[cl428_cl434GDI<span class="sc">$</span>sum.raw.norm <span class="sc">&gt;</span> <span class="dv">7</span>,]</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>top.GDI.genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(subsetGDIcl428_cl434[<span class="fu">order</span>(subsetGDIcl428_cl434<span class="sc">$</span>GDI,<span class="at">decreasing =</span> T),])[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>]</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">GDIPlot</span>(cl428_cl434,<span class="at">genes =</span> <span class="st">""</span>,<span class="at">GDIIn =</span> cl428_cl434GDI)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Real_UT_clustersByCotan_files/figure-html/unnamed-chunk-2-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">getNormalizedData</span>(cl428_cl434)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> data[<span class="sc">!</span><span class="fu">rowSums</span>(<span class="fu">as.matrix</span>(data)) <span class="sc">&lt;</span> <span class="dv">1</span>,]</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">log</span>(data<span class="sc">*</span><span class="dv">10000</span><span class="sc">+</span><span class="dv">1</span>)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>row_stdev <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">1</span>, sd, <span class="at">na.rm=</span><span class="cn">TRUE</span>)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>row_stdev <span class="ot">&lt;-</span> row_stdev[<span class="fu">order</span>(row_stdev,<span class="at">decreasing =</span> T)]</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>genes.to.keep <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">names</span>(row_stdev[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>]),top.GDI.genes)</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>data.small <span class="ot">&lt;-</span> data[<span class="fu">rownames</span>(data) <span class="sc">%in%</span> genes.to.keep,]</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="co">#data &lt;- t(as.matrix(data))</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>data.small <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">as.matrix</span>(data.small))</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>COTAN_Cl.code <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">getClusterizationData</span>(cl428_cl434,<span class="at">clName =</span> <span class="st">"original.clusters"</span>)[[<span class="dv">1</span>]])</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>COTAN_Cl.code <span class="ot">&lt;-</span> COTAN_Cl.code <span class="sc">-</span><span class="dv">1</span></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>data.small <span class="ot">&lt;-</span> <span class="fu">cbind</span>(data.small,COTAN_Cl.code)</span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>data.small <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(data.small)</span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Split the data into training and test set</span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>training.samples <span class="ot">&lt;-</span> data.small[,<span class="st">"COTAN_Cl.code"</span>] <span class="sc">%&gt;%</span> </span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">createDataPartition</span>(<span class="at">p =</span> <span class="fl">0.8</span>, <span class="at">list =</span> <span class="cn">FALSE</span>)</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>train.data  <span class="ot">&lt;-</span> data.small[training.samples, ]</span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a>test.data <span class="ot">&lt;-</span> data.small[<span class="sc">-</span>training.samples, ]</span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(train.data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                               Abracl  Cdkn1c    Cited2     Mllt3  Fam210b
10X74_4_A_1:GTAACGTGCCATAGx 10.128836 0.00000  9.435728  9.435728 0.000000
10X73_3_A_1:ACGCCGGATCGTAGx 10.211198 0.00000 10.904327 10.211198 9.518087
10X74_4_A_1:GATTCGGAGGTGAGx 10.518544 9.13233  9.825424  9.132330 9.132330
10X74_4_A_1:CTGAGCCTGGTAAAx 10.916737 0.00000  9.307371 10.405923 9.307371
10X74_4_A_1:CCTAGAGAGTCACAx  9.585177 0.00000 11.194560  0.000000 0.000000
10X74_4_A_1:CTAGGATGCTTGCCx 11.828628 0.00000 10.941336  8.995532 8.995532
                                Mfap4   Cxcl12  Cdc42ep3    Limch1     Ier2
10X74_4_A_1:GTAACGTGCCATAGx  9.435728 9.435728  0.000000  0.000000 0.000000
10X73_3_A_1:ACGCCGGATCGTAGx  9.518087 0.000000  9.518087  0.000000 0.000000
10X74_4_A_1:GATTCGGAGGTGAGx 11.078148 9.825424  0.000000  9.132330 9.132330
10X74_4_A_1:CTGAGCCTGGTAAAx 10.405923 0.000000 10.000473 10.000473 9.307371
10X74_4_A_1:CCTAGAGAGTCACAx  9.585177 9.585177  0.000000  0.000000 0.000000
10X74_4_A_1:CTAGGATGCTTGCCx 10.941336 9.688617  0.000000  8.995532 8.995532
                                Ptprk   Gm29260    Sfrp1    Pantr1     Magi2
10X74_4_A_1:GTAACGTGCCATAGx  9.435728  0.000000 9.435728 10.534287  0.000000
10X73_3_A_1:ACGCCGGATCGTAGx 10.211198  9.518087 9.518087  9.518087  9.518087
10X74_4_A_1:GATTCGGAGGTGAGx  9.132330  9.825424 9.825424 10.518544  0.000000
10X74_4_A_1:CTGAGCCTGGTAAAx  9.307371 10.000473 0.000000 10.405923  0.000000
10X74_4_A_1:CCTAGAGAGTCACAx  9.585177  0.000000 9.585177  0.000000  0.000000
10X74_4_A_1:CTAGGATGCTTGCCx  0.000000  8.995532 9.688617 10.604871 10.381733
                             Tmem108     Ddr1       Ezr    Sncaip CT025619.1
10X74_4_A_1:GTAACGTGCCATAGx 0.000000 0.000000  9.435728  0.000000   9.435728
10X73_3_A_1:ACGCCGGATCGTAGx 0.000000 0.000000  0.000000  9.518087  10.211198
10X74_4_A_1:GATTCGGAGGTGAGx 0.000000 0.000000 10.230871  9.825424   9.132330
10X74_4_A_1:CTGAGCCTGGTAAAx 9.307371 0.000000  0.000000 10.000473   9.307371
10X74_4_A_1:CCTAGAGAGTCACAx 0.000000 0.000000 10.278290  0.000000   9.585177
10X74_4_A_1:CTAGGATGCTTGCCx 8.995532 8.995532  8.995532  0.000000   8.995532
                               Fabp7 Ctnnd2       Ptn    Frmd4a    Sorbs2
10X74_4_A_1:GTAACGTGCCATAGx 0.000000      0 10.128836 11.227421 10.534287
10X73_3_A_1:ACGCCGGATCGTAGx 9.518087      0  9.518087  9.518087 10.616651
10X74_4_A_1:GATTCGGAGGTGAGx 9.132330      0  0.000000  0.000000  9.825424
10X74_4_A_1:CTGAGCCTGGTAAAx 9.307371      0 10.693598  9.307371 10.000473
10X74_4_A_1:CCTAGAGAGTCACAx 9.585177      0 10.278290  0.000000  9.585177
10X74_4_A_1:CTAGGATGCTTGCCx 9.688617      0  0.000000  0.000000 10.094061
                                Hes6       Dcc   Neurog2     Tcf12  Fam110a
10X74_4_A_1:GTAACGTGCCATAGx 0.000000  9.435728  0.000000  0.000000 0.000000
10X73_3_A_1:ACGCCGGATCGTAGx 9.518087  0.000000  0.000000 10.616651 0.000000
10X74_4_A_1:GATTCGGAGGTGAGx 9.825424  9.825424 10.741682  9.825424 9.825424
10X74_4_A_1:CTGAGCCTGGTAAAx 0.000000 10.916737  0.000000 10.000473 0.000000
10X74_4_A_1:CCTAGAGAGTCACAx 9.585177  0.000000  0.000000 11.194560 9.585177
10X74_4_A_1:CTAGGATGCTTGCCx 0.000000  8.995532  8.995532 10.094061 0.000000
                               Dhrs4     Ccnd2    Elavl4      Ank3    Klf12
10X74_4_A_1:GTAACGTGCCATAGx 0.000000  0.000000  9.435728  9.435728 0.000000
10X73_3_A_1:ACGCCGGATCGTAGx 0.000000  9.518087  0.000000  9.518087 0.000000
10X74_4_A_1:GATTCGGAGGTGAGx 9.132330 10.924000  9.132330  0.000000 9.132330
10X74_4_A_1:CTGAGCCTGGTAAAx 0.000000  0.000000 10.000473 10.916737 9.307371
10X74_4_A_1:CCTAGAGAGTCACAx 0.000000 10.278290 10.971420  9.585177 9.585177
10X74_4_A_1:CTAGGATGCTTGCCx 9.688617 10.094061 10.094061  9.688617 0.000000
                             Slc17a6     Lzts1      Map2      Lhx2   Neurod2
10X74_4_A_1:GTAACGTGCCATAGx 0.000000  0.000000  0.000000 10.128836 10.534287
10X73_3_A_1:ACGCCGGATCGTAGx 0.000000  9.518087 10.211198  9.518087  9.518087
10X74_4_A_1:GATTCGGAGGTGAGx 9.132330 10.518544  9.825424 10.230871  9.825424
10X74_4_A_1:CTGAGCCTGGTAAAx 0.000000 10.405923  0.000000 10.000473 11.099055
10X74_4_A_1:CCTAGAGAGTCACAx 0.000000  0.000000 10.971420 10.971420 10.278290
10X74_4_A_1:CTAGGATGCTTGCCx 8.995532 10.381733  8.995532 10.787188 10.604871
                               Sstr2     Eomes   Vps37b      Ier5  Cttnbp2
10X74_4_A_1:GTAACGTGCCATAGx 10.82196  0.000000  0.00000 10.128836  0.00000
10X73_3_A_1:ACGCCGGATCGTAGx 10.21120  0.000000  0.00000  9.518087 11.12747
10X74_4_A_1:GATTCGGAGGTGAGx 10.51854  9.825424  9.13233  0.000000 10.23087
10X74_4_A_1:CTGAGCCTGGTAAAx 10.69360  0.000000  0.00000  0.000000  0.00000
10X74_4_A_1:CCTAGAGAGTCACAx 10.27829 10.278290 10.27829  9.585177 11.19456
10X74_4_A_1:CTAGGATGCTTGCCx 10.94134  0.000000 10.60487  0.000000 10.94134
                              Tmem178   Plxna4      Aff3     Tiam2  Ppp2r2b
10X74_4_A_1:GTAACGTGCCATAGx  9.435728 9.435728  9.435728  9.435728 10.82196
10X73_3_A_1:ACGCCGGATCGTAGx  0.000000 0.000000 10.616651 10.211198 10.21120
10X74_4_A_1:GATTCGGAGGTGAGx  9.132330 0.000000  0.000000 10.230871  9.13233
10X74_4_A_1:CTGAGCCTGGTAAAx 10.405923 0.000000  0.000000  9.307371 10.91674
10X74_4_A_1:CCTAGAGAGTCACAx  0.000000 0.000000  0.000000  0.000000 10.27829
10X74_4_A_1:CTAGGATGCTTGCCx  0.000000 0.000000  0.000000 10.094061 10.09406
                                 Zeb2     Gria2     Nbea     Ptprd      Sox5
10X74_4_A_1:GTAACGTGCCATAGx  9.435728 10.534287 0.000000  9.435728 10.821963
10X73_3_A_1:ACGCCGGATCGTAGx  0.000000 10.211198 9.518087  0.000000  9.518087
10X74_4_A_1:GATTCGGAGGTGAGx  9.825424  9.132330 0.000000  0.000000  9.132330
10X74_4_A_1:CTGAGCCTGGTAAAx  0.000000 10.405923 0.000000 10.405923 10.000473
10X74_4_A_1:CCTAGAGAGTCACAx  9.585177  9.585177 0.000000  0.000000 10.278290
10X74_4_A_1:CTAGGATGCTTGCCx 10.381733  0.000000 0.000000  0.000000  0.000000
                               Mpped2   Bcl11b    Epha5     Nol4    Grik2
10X74_4_A_1:GTAACGTGCCATAGx 10.534287 9.435728 10.53429  0.00000 9.435728
10X73_3_A_1:ACGCCGGATCGTAGx 10.211198 0.000000 10.61665 10.21120 0.000000
10X74_4_A_1:GATTCGGAGGTGAGx 10.924000 0.000000 10.23087  0.00000 9.132330
10X74_4_A_1:CTGAGCCTGGTAAAx  9.307371 0.000000 10.00047 10.91674 0.000000
10X74_4_A_1:CCTAGAGAGTCACAx  0.000000 0.000000 10.68374 10.97142 0.000000
10X74_4_A_1:CTAGGATGCTTGCCx  9.688617 0.000000 10.38173  0.00000 8.995532
                            Mir124.2hg      Pcp4   Ccser1     Kcnb2   Dscaml1
10X74_4_A_1:GTAACGTGCCATAGx   0.000000  0.000000 11.63288 10.128836  0.000000
10X73_3_A_1:ACGCCGGATCGTAGx   0.000000 11.127466 10.90433  0.000000  0.000000
10X74_4_A_1:GATTCGGAGGTGAGx   0.000000 10.741682  0.00000  9.132330  9.132330
10X74_4_A_1:CTGAGCCTGGTAAAx   0.000000  0.000000 11.09906  0.000000  0.000000
10X74_4_A_1:CCTAGAGAGTCACAx   0.000000 10.971420 10.27829  0.000000 10.278290
10X74_4_A_1:CTAGGATGCTTGCCx   8.995532  9.688617 10.38173  8.995532  9.688617
                                Kcnh7       Pam     Negr1     Ppm1e     Myt1l
10X74_4_A_1:GTAACGTGCCATAGx  0.000000  0.000000  9.435728 10.534287 11.515100
10X73_3_A_1:ACGCCGGATCGTAGx  0.000000 10.616651 10.616651  9.518087 10.211198
10X74_4_A_1:GATTCGGAGGTGAGx  9.132330  9.825424 10.518544  9.825424  0.000000
10X74_4_A_1:CTGAGCCTGGTAAAx  9.307371  9.307371  0.000000  0.000000 10.405923
10X74_4_A_1:CCTAGAGAGTCACAx  9.585177 10.683744 11.194560  0.000000  9.585177
10X74_4_A_1:CTAGGATGCTTGCCx 10.381733  8.995532  0.000000  9.688617 10.094061
                                Foxp2     Gpc6     Masp1 Serpini1    Pcdh7
10X74_4_A_1:GTAACGTGCCATAGx  9.435728 10.12884  0.000000 9.435728 10.12884
10X73_3_A_1:ACGCCGGATCGTAGx  9.518087 10.21120  9.518087 0.000000 11.12747
10X74_4_A_1:GATTCGGAGGTGAGx  0.000000  0.00000  9.132330 0.000000  9.13233
10X74_4_A_1:CTGAGCCTGGTAAAx 10.405923  0.00000 10.693598 0.000000  0.00000
10X74_4_A_1:CCTAGAGAGTCACAx  0.000000  0.00000  0.000000 0.000000 10.97142
10X74_4_A_1:CTAGGATGCTTGCCx  0.000000  0.00000  9.688617 0.000000 10.09406
                               Rbfox1    Robo2     Fhod3     Rprm      Gng3
10X74_4_A_1:GTAACGTGCCATAGx 10.821963 11.38157  9.435728 0.000000 10.128836
10X73_3_A_1:ACGCCGGATCGTAGx 11.463934 11.71525 10.211198 0.000000  0.000000
10X74_4_A_1:GATTCGGAGGTGAGx  9.132330 11.32946  9.132330 9.825424  9.132330
10X74_4_A_1:CTGAGCCTGGTAAAx 11.099055 10.69360  0.000000 0.000000  0.000000
10X74_4_A_1:CCTAGAGAGTCACAx  9.585177 10.68374  0.000000 0.000000  0.000000
10X74_4_A_1:CTAGGATGCTTGCCx  0.000000 11.19265  9.688617 8.995532  8.995532
                               Uchl1  Tmem176b      Nrn1   Snap25     Nrg1
10X74_4_A_1:GTAACGTGCCATAGx 0.000000  9.435728  0.000000 9.435728 11.04510
10X73_3_A_1:ACGCCGGATCGTAGx 0.000000 10.211198  0.000000 0.000000 10.90433
10X74_4_A_1:GATTCGGAGGTGAGx 0.000000  9.132330  9.825424 0.000000 10.74168
10X74_4_A_1:CTGAGCCTGGTAAAx 9.307371  0.000000 10.916737 9.307371 11.38673
10X74_4_A_1:CCTAGAGAGTCACAx 0.000000  9.585177  9.585177 9.585177 10.97142
10X74_4_A_1:CTAGGATGCTTGCCx 0.000000  8.995532  9.688617 9.688617 11.19265
                                Cdh8   Trim17     Chga    Rpl26     Rack1
10X74_4_A_1:GTAACGTGCCATAGx 10.12884 0.000000 0.000000 10.82196  0.000000
10X73_3_A_1:ACGCCGGATCGTAGx 11.30979 0.000000 9.518087 11.30979  0.000000
10X74_4_A_1:GATTCGGAGGTGAGx 10.74168 0.000000 0.000000 10.92400 10.230871
10X74_4_A_1:CTGAGCCTGGTAAAx 10.40592 9.307371 9.307371 10.91674  9.307371
10X74_4_A_1:CCTAGAGAGTCACAx 10.27829 9.585177 0.000000 10.68374 10.278290
10X74_4_A_1:CTAGGATGCTTGCCx 10.38173 8.995532 0.000000 11.29801 10.094061
                             Hsp90b1    Aldoa   Tuba1b    Sf3b6      Top1
10X74_4_A_1:GTAACGTGCCATAGx 9.435728 9.435728 9.435728  0.00000 10.534287
10X73_3_A_1:ACGCCGGATCGTAGx 0.000000 9.518087 9.518087  0.00000  9.518087
10X74_4_A_1:GATTCGGAGGTGAGx 0.000000 9.132330 9.132330  9.13233 10.230871
10X74_4_A_1:CTGAGCCTGGTAAAx 0.000000 9.307371 9.307371  0.00000  9.307371
10X74_4_A_1:CCTAGAGAGTCACAx 0.000000 0.000000 0.000000  0.00000  9.585177
10X74_4_A_1:CTAGGATGCTTGCCx 0.000000 8.995532 0.000000 10.09406 10.094061
                                  Tpr     Smc3     Rplp1     Rpl22   Rpl36a
10X74_4_A_1:GTAACGTGCCATAGx  9.435728 0.000000 10.128836  0.000000 10.53429
10X73_3_A_1:ACGCCGGATCGTAGx  0.000000 0.000000 10.616651  0.000000  0.00000
10X74_4_A_1:GATTCGGAGGTGAGx 10.230871 0.000000 10.518544  9.132330  0.00000
10X74_4_A_1:CTGAGCCTGGTAAAx  0.000000 0.000000  9.307371 10.000473 11.09906
10X74_4_A_1:CCTAGAGAGTCACAx  9.585177 0.000000 10.683744  0.000000 10.68374
10X74_4_A_1:CTAGGATGCTTGCCx  0.000000 8.995532 10.094061  9.688617 10.09406
                                Rps21    Eif4a1      Erh      Mdk     Pde4d
10X74_4_A_1:GTAACGTGCCATAGx  9.435728  9.435728 0.000000  0.00000  9.435728
10X73_3_A_1:ACGCCGGATCGTAGx  0.000000 10.211198 0.000000 10.61665 10.616651
10X74_4_A_1:GATTCGGAGGTGAGx 10.741682  9.825424 0.000000 11.07815  0.000000
10X74_4_A_1:CTGAGCCTGGTAAAx  9.307371 10.916737 0.000000 11.79219  0.000000
10X74_4_A_1:CCTAGAGAGTCACAx  9.585177 10.278290 0.000000 11.19456  9.585177
10X74_4_A_1:CTAGGATGCTTGCCx 10.381733  9.688617 9.688617 10.94134 10.941336
                              Pgrmc1    Ube2r2     Bri3     Rps28     Mbtd1
10X74_4_A_1:GTAACGTGCCATAGx 0.000000  0.000000 0.000000  9.435728  0.000000
10X73_3_A_1:ACGCCGGATCGTAGx 0.000000  9.518087 0.000000  9.518087 10.616651
10X74_4_A_1:GATTCGGAGGTGAGx 9.132330  0.000000 9.132330  9.132330  0.000000
10X74_4_A_1:CTGAGCCTGGTAAAx 9.307371 10.000473 9.307371 10.000473  9.307371
10X74_4_A_1:CCTAGAGAGTCACAx 9.585177  0.000000 9.585177  0.000000 10.683744
10X74_4_A_1:CTAGGATGCTTGCCx 0.000000  0.000000 9.688617  8.995532  9.688617
                               Rpl15     Macf1   Phf21a     Oaz2      Gnb1
10X74_4_A_1:GTAACGTGCCATAGx 10.53429  9.435728 0.000000 10.12884  0.000000
10X73_3_A_1:ACGCCGGATCGTAGx  0.00000  9.518087 0.000000  0.00000 10.211198
10X74_4_A_1:GATTCGGAGGTGAGx 10.51854  0.000000 9.825424  9.13233  0.000000
10X74_4_A_1:CTGAGCCTGGTAAAx 10.40592  9.307371 0.000000  0.00000  9.307371
10X74_4_A_1:CCTAGAGAGTCACAx 10.68374  0.000000 0.000000  0.00000  9.585177
10X74_4_A_1:CTAGGATGCTTGCCx 10.60487 10.381733 8.995532  0.00000  0.000000
                               Zfp422    Rpl35a     Cox8a   Nt5dc2
10X74_4_A_1:GTAACGTGCCATAGx 10.128836 10.821963 10.128836  0.00000
10X73_3_A_1:ACGCCGGATCGTAGx  0.000000  9.518087 10.211198  0.00000
10X74_4_A_1:GATTCGGAGGTGAGx  9.132330 10.518544  9.132330 10.92400
10X74_4_A_1:CTGAGCCTGGTAAAx  0.000000 10.000473  9.307371 10.00047
10X74_4_A_1:CCTAGAGAGTCACAx  0.000000  0.000000  0.000000  0.00000
10X74_4_A_1:CTAGGATGCTTGCCx  8.995532  9.688617  8.995532 10.09406
                            X4930402H24Rik     Ttc28      Ckb   Sh3bgrl
10X74_4_A_1:GTAACGTGCCATAGx      10.821963 10.128836 9.435728  0.000000
10X73_3_A_1:ACGCCGGATCGTAGx       9.518087 10.616651 0.000000  9.518087
10X74_4_A_1:GATTCGGAGGTGAGx       0.000000 10.741682 9.132330  0.000000
10X74_4_A_1:CTGAGCCTGGTAAAx      10.000473 11.253204 0.000000  0.000000
10X74_4_A_1:CCTAGAGAGTCACAx       0.000000 10.278290 0.000000 10.278290
10X74_4_A_1:CTAGGATGCTTGCCx       0.000000  9.688617 8.995532  9.688617
                               Top2b  Ndufa10 C530008M17Rik     Basp1  Gpatch8
10X74_4_A_1:GTAACGTGCCATAGx 9.435728  0.00000      0.000000 10.534287 9.435728
10X73_3_A_1:ACGCCGGATCGTAGx 0.000000  0.00000      0.000000 10.616651 9.518087
10X74_4_A_1:GATTCGGAGGTGAGx 0.000000 10.23087      0.000000 10.518544 9.132330
10X74_4_A_1:CTGAGCCTGGTAAAx 0.000000 10.00047      0.000000  9.307371 9.307371
10X74_4_A_1:CCTAGAGAGTCACAx 0.000000  0.00000      9.585177  0.000000 0.000000
10X74_4_A_1:CTAGGATGCTTGCCx 8.995532  0.00000     10.094061  9.688617 8.995532
                                Map1b  Ccdc88a     Xist    Ctnna2     Rnf7
10X74_4_A_1:GTAACGTGCCATAGx 10.534287 0.000000  0.00000 10.534287 9.435728
10X73_3_A_1:ACGCCGGATCGTAGx  9.518087 9.518087 10.61665 11.463934 0.000000
10X74_4_A_1:GATTCGGAGGTGAGx  0.000000 0.000000  0.00000  9.825424 0.000000
10X74_4_A_1:CTGAGCCTGGTAAAx 10.000473 0.000000 10.69360 10.000473 9.307371
10X74_4_A_1:CCTAGAGAGTCACAx  9.585177 0.000000 11.78234  0.000000 0.000000
10X74_4_A_1:CTAGGATGCTTGCCx  0.000000 8.995532 10.38173  9.688617 9.688617
                                Fabp5    Usmg5     Phip     Rbmx    Zbtb20
10X74_4_A_1:GTAACGTGCCATAGx  0.000000 9.435728 9.435728 10.12884  9.435728
10X73_3_A_1:ACGCCGGATCGTAGx  9.518087 0.000000 0.000000  0.00000 10.211198
10X74_4_A_1:GATTCGGAGGTGAGx  9.825424 0.000000 9.132330  9.13233 10.518544
10X74_4_A_1:CTGAGCCTGGTAAAx 10.000473 9.307371 0.000000  0.00000  9.307371
10X74_4_A_1:CCTAGAGAGTCACAx  0.000000 0.000000 0.000000  0.00000 10.278290
10X74_4_A_1:CTAGGATGCTTGCCx 10.381733 8.995532 0.000000  0.00000  8.995532
                                Eif1b      Lcor      Nsg1      Evl      Meg3
10X74_4_A_1:GTAACGTGCCATAGx  9.435728  0.000000  9.435728 9.435728  0.000000
10X73_3_A_1:ACGCCGGATCGTAGx 10.616651  9.518087  9.518087 9.518087  0.000000
10X74_4_A_1:GATTCGGAGGTGAGx  9.132330  9.825424  9.825424 0.000000  0.000000
10X74_4_A_1:CTGAGCCTGGTAAAx 10.693598 10.000473 10.693598 0.000000 10.000473
10X74_4_A_1:CCTAGAGAGTCACAx  9.585177  0.000000 10.683744 0.000000  9.585177
10X74_4_A_1:CTAGGATGCTTGCCx 10.094061  0.000000  9.688617 8.995532  8.995532
                               Eif4a2     Rab10    Mllt11 COTAN_Cl.code
10X74_4_A_1:GTAACGTGCCATAGx 10.821963  0.000000 10.128836             0
10X73_3_A_1:ACGCCGGATCGTAGx  0.000000  9.518087  0.000000             0
10X74_4_A_1:GATTCGGAGGTGAGx  0.000000 10.230871  9.132330             0
10X74_4_A_1:CTGAGCCTGGTAAAx  0.000000  0.000000 10.405923             0
10X74_4_A_1:CCTAGAGAGTCACAx  9.585177  0.000000  9.585177             0
10X74_4_A_1:CTAGGATGCTTGCCx  0.000000  0.000000 10.381733             0</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Fit the model</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="fu">glm</span>( COTAN_Cl.code <span class="sc">~</span>., <span class="at">data =</span> train.data, <span class="at">family =</span> binomial,<span class="at">control =</span> <span class="fu">list</span>(<span class="at">maxit =</span> <span class="dv">50</span>))</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize the model</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(model)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
glm(formula = COTAN_Cl.code ~ ., family = binomial, data = train.data, 
    control = list(maxit = 50))

Coefficients:
                 Estimate Std. Error    z value Pr(&gt;|z|)    
(Intercept)    -1.392e+15  5.418e+07  -25692175   &lt;2e-16 ***
Abracl         -3.668e+13  1.108e+06  -33117531   &lt;2e-16 ***
Cdkn1c          9.057e+13  8.449e+05  107200885   &lt;2e-16 ***
Cited2         -1.192e+14  1.158e+06 -102925237   &lt;2e-16 ***
Mllt3          -1.411e+12  9.271e+05   -1522016   &lt;2e-16 ***
Fam210b        -4.090e+12  7.886e+05   -5185950   &lt;2e-16 ***
Mfap4          -6.719e+12  1.105e+06   -6082731   &lt;2e-16 ***
Cxcl12          3.911e+13  8.228e+05   47537816   &lt;2e-16 ***
Cdc42ep3        1.404e+13  8.102e+05   17326401   &lt;2e-16 ***
Limch1          4.103e+13  7.629e+05   53781940   &lt;2e-16 ***
Ier2           -1.632e+13  8.096e+05  -20155968   &lt;2e-16 ***
Ptprk          -3.990e+13  7.877e+05  -50648051   &lt;2e-16 ***
Gm29260        -1.591e+14  8.513e+05 -186868495   &lt;2e-16 ***
Sfrp1          -1.001e+13  7.665e+05  -13056648   &lt;2e-16 ***
Pantr1         -4.908e+13  1.184e+06  -41449633   &lt;2e-16 ***
Magi2           2.042e+13  7.718e+05   26452066   &lt;2e-16 ***
Tmem108         3.884e+13  8.406e+05   46203169   &lt;2e-16 ***
Ddr1           -6.061e+13  7.777e+05  -77935451   &lt;2e-16 ***
Ezr            -1.538e+12  8.165e+05   -1883070   &lt;2e-16 ***
Sncaip         -6.607e+13  7.615e+05  -86758312   &lt;2e-16 ***
CT025619.1      5.786e+13  7.693e+05   75220795   &lt;2e-16 ***
Fabp7          -1.160e+13  7.987e+05  -14523404   &lt;2e-16 ***
Ctnnd2          5.243e+13  7.989e+05   65626431   &lt;2e-16 ***
Ptn            -5.180e+13  8.866e+05  -58423528   &lt;2e-16 ***
Frmd4a         -9.865e+12  8.139e+05  -12119603   &lt;2e-16 ***
Sorbs2          7.082e+13  1.040e+06   68115662   &lt;2e-16 ***
Hes6            5.248e+13  8.207e+05   63952536   &lt;2e-16 ***
Dcc            -1.086e+14  1.091e+06  -99558297   &lt;2e-16 ***
Neurog2         8.885e+12  8.991e+05    9882478   &lt;2e-16 ***
Tcf12           4.439e+13  7.580e+05   58560212   &lt;2e-16 ***
Fam110a        -3.762e+12  7.726e+05   -4870098   &lt;2e-16 ***
Dhrs4          -6.585e+13  8.280e+05  -79533727   &lt;2e-16 ***
Ccnd2           2.088e+13  8.477e+05   24635620   &lt;2e-16 ***
Elavl4          5.882e+13  1.040e+06   56574947   &lt;2e-16 ***
Ank3            1.621e+12  8.158e+05    1986603   &lt;2e-16 ***
Klf12           1.858e+13  8.113e+05   22905550   &lt;2e-16 ***
Slc17a6         3.336e+13  7.777e+05   42892923   &lt;2e-16 ***
Lzts1          -3.340e+11  7.629e+05    -437779   &lt;2e-16 ***
Map2            1.529e+13  1.049e+06   14565916   &lt;2e-16 ***
Lhx2            6.076e+13  1.073e+06   56647519   &lt;2e-16 ***
Neurod2         1.156e+14  1.308e+06   88402243   &lt;2e-16 ***
Sstr2           2.958e+13  1.123e+06   26328080   &lt;2e-16 ***
Eomes           2.323e+13  8.998e+05   25818313   &lt;2e-16 ***
Vps37b          2.378e+13  7.719e+05   30806540   &lt;2e-16 ***
Ier5           -6.635e+13  7.933e+05  -83638912   &lt;2e-16 ***
Cttnbp2         7.413e+12  1.026e+06    7226886   &lt;2e-16 ***
Tmem178         5.869e+13  7.711e+05   76112402   &lt;2e-16 ***
Plxna4          6.142e+12  8.247e+05    7447401   &lt;2e-16 ***
Aff3            3.216e+13  8.494e+05   37865603   &lt;2e-16 ***
Tiam2           3.504e+13  8.431e+05   41564359   &lt;2e-16 ***
Ppp2r2b         1.014e+12  1.343e+06     755110   &lt;2e-16 ***
Zeb2           -8.426e+13  8.214e+05 -102580264   &lt;2e-16 ***
Gria2           3.458e+13  8.115e+05   42614466   &lt;2e-16 ***
Nbea           -2.142e+13  7.864e+05  -27232237   &lt;2e-16 ***
Ptprd          -8.085e+12  8.432e+05   -9588010   &lt;2e-16 ***
Sox5            1.043e+13  9.521e+05   10952629   &lt;2e-16 ***
Mpped2          1.653e+13  9.316e+05   17741622   &lt;2e-16 ***
Bcl11b         -5.000e+13  8.186e+05  -61072633   &lt;2e-16 ***
Epha5           7.168e+13  1.438e+06   49838920   &lt;2e-16 ***
Nol4            5.393e+13  8.773e+05   61475929   &lt;2e-16 ***
Grik2          -8.502e+12  7.828e+05  -10860942   &lt;2e-16 ***
Mir124.2hg      3.667e+12  8.306e+05    4415594   &lt;2e-16 ***
Pcp4           -9.935e+12  7.703e+05  -12897946   &lt;2e-16 ***
Ccser1          6.731e+13  1.011e+06   66545181   &lt;2e-16 ***
Kcnb2          -3.395e+13  7.809e+05  -43477366   &lt;2e-16 ***
Dscaml1        -3.299e+13  8.061e+05  -40923228   &lt;2e-16 ***
Kcnh7          -3.625e+13  8.216e+05  -44124624   &lt;2e-16 ***
Pam            -2.266e+13  9.566e+05  -23682478   &lt;2e-16 ***
Negr1          -5.486e+13  8.125e+05  -67527681   &lt;2e-16 ***
Ppm1e          -1.101e+13  7.777e+05  -14161188   &lt;2e-16 ***
Myt1l           1.275e+13  9.473e+05   13455861   &lt;2e-16 ***
Foxp2          -4.217e+12  7.833e+05   -5382953   &lt;2e-16 ***
Gpc6           -7.684e+12  7.945e+05   -9671234   &lt;2e-16 ***
Masp1           4.292e+13  7.665e+05   55993994   &lt;2e-16 ***
Serpini1       -5.105e+12  8.061e+05   -6333133   &lt;2e-16 ***
Pcdh7           5.837e+13  8.073e+05   72299867   &lt;2e-16 ***
Rbfox1          7.461e+12  8.031e+05    9289560   &lt;2e-16 ***
Robo2          -6.876e+13  1.420e+06  -48418494   &lt;2e-16 ***
Fhod3           4.399e+13  7.980e+05   55121613   &lt;2e-16 ***
Rprm            3.484e+13  7.835e+05   44465376   &lt;2e-16 ***
Gng3            2.281e+13  7.680e+05   29706841   &lt;2e-16 ***
Uchl1          -1.332e+13  7.594e+05  -17544470   &lt;2e-16 ***
Tmem176b        8.467e+12  8.030e+05   10544314   &lt;2e-16 ***
Nrn1           -5.681e+13  9.557e+05  -59447760   &lt;2e-16 ***
Snap25         -4.422e+13  7.898e+05  -55987310   &lt;2e-16 ***
Nrg1           -3.190e+13  1.511e+06  -21119116   &lt;2e-16 ***
Cdh8           -9.450e+13  9.459e+05  -99906199   &lt;2e-16 ***
Trim17         -5.827e+13  7.920e+05  -73573697   &lt;2e-16 ***
Chga            1.515e+13  7.864e+05   19260283   &lt;2e-16 ***
Rpl26          -1.381e+14  1.158e+06 -119242819   &lt;2e-16 ***
Rack1          -3.030e+13  1.065e+06  -28463399   &lt;2e-16 ***
Hsp90b1         1.513e+12  7.764e+05    1949079   &lt;2e-16 ***
Aldoa           5.999e+13  7.895e+05   75985414   &lt;2e-16 ***
Tuba1b         -2.984e+13  7.673e+05  -38893858   &lt;2e-16 ***
Sf3b6          -1.751e+13  7.754e+05  -22583944   &lt;2e-16 ***
Top1           -4.635e+13  1.256e+06  -36914861   &lt;2e-16 ***
Tpr             2.535e+13  8.111e+05   31258405   &lt;2e-16 ***
Smc3            1.778e+13  7.864e+05   22614692   &lt;2e-16 ***
Rplp1           6.245e+13  1.506e+06   41453287   &lt;2e-16 ***
Rpl22           5.820e+13  9.818e+05   59284313   &lt;2e-16 ***
Rpl36a          7.624e+13  1.286e+06   59290521   &lt;2e-16 ***
Rps21          -1.722e+13  9.803e+05  -17562653   &lt;2e-16 ***
Eif4a1          3.635e+13  1.107e+06   32824725   &lt;2e-16 ***
Erh             3.879e+13  8.249e+05   47018259   &lt;2e-16 ***
Mdk             6.137e+13  1.311e+06   46812498   &lt;2e-16 ***
Pde4d           9.697e+13  8.054e+05  120403478   &lt;2e-16 ***
Pgrmc1          4.160e+13  7.890e+05   52731940   &lt;2e-16 ***
Ube2r2         -4.281e+12  7.630e+05   -5610419   &lt;2e-16 ***
Bri3           -4.674e+12  7.624e+05   -6130738   &lt;2e-16 ***
Rps28           2.337e+13  1.081e+06   21624714   &lt;2e-16 ***
Mbtd1          -3.431e+13  8.020e+05  -42787850   &lt;2e-16 ***
Rpl15           4.540e+13  1.056e+06   42976360   &lt;2e-16 ***
Macf1          -3.898e+13  7.456e+05  -52271693   &lt;2e-16 ***
Phf21a         -1.085e+13  7.941e+05  -13663454   &lt;2e-16 ***
Oaz2           -2.184e+13  7.671e+05  -28476312   &lt;2e-16 ***
Gnb1            3.086e+13  7.786e+05   39638926   &lt;2e-16 ***
Zfp422         -2.665e+13  7.937e+05  -33579867   &lt;2e-16 ***
Rpl35a         -4.777e+13  1.012e+06  -47191386   &lt;2e-16 ***
Cox8a           3.798e+12  9.384e+05    4047518   &lt;2e-16 ***
Nt5dc2         -2.355e+13  7.820e+05  -30114999   &lt;2e-16 ***
X4930402H24Rik -3.421e+13  8.011e+05  -42703505   &lt;2e-16 ***
Ttc28          -2.613e+13  9.395e+05  -27807041   &lt;2e-16 ***
Ckb            -4.380e+13  8.412e+05  -52071863   &lt;2e-16 ***
Sh3bgrl        -1.422e+13  7.831e+05  -18158721   &lt;2e-16 ***
Top2b           4.336e+13  7.850e+05   55234013   &lt;2e-16 ***
Ndufa10        -8.308e+13  7.998e+05 -103871878   &lt;2e-16 ***
C530008M17Rik  -4.875e+13  8.134e+05  -59928713   &lt;2e-16 ***
Basp1           1.084e+14  1.002e+06  108114394   &lt;2e-16 ***
Gpatch8         4.789e+12  7.653e+05    6257168   &lt;2e-16 ***
Map1b          -3.338e+13  9.207e+05  -36254006   &lt;2e-16 ***
Ccdc88a         9.193e+12  7.736e+05   11882326   &lt;2e-16 ***
Xist           -2.576e+13  7.861e+05  -32765738   &lt;2e-16 ***
Ctnna2         -4.233e+13  9.718e+05  -43552178   &lt;2e-16 ***
Rnf7           -5.865e+11  8.067e+05    -727059   &lt;2e-16 ***
Fabp5           6.801e+13  9.924e+05   68526316   &lt;2e-16 ***
Usmg5           3.478e+13  7.931e+05   43858152   &lt;2e-16 ***
Phip            3.203e+13  7.823e+05   40942232   &lt;2e-16 ***
Rbmx            5.754e+12  7.791e+05    7384674   &lt;2e-16 ***
Zbtb20         -4.798e+13  1.022e+06  -46936751   &lt;2e-16 ***
Eif1b          -7.349e+13  1.019e+06  -72109569   &lt;2e-16 ***
Lcor            1.574e+13  7.874e+05   19989987   &lt;2e-16 ***
Nsg1           -8.545e+12  9.945e+05   -8592451   &lt;2e-16 ***
Evl            -1.350e+13  7.905e+05  -17079400   &lt;2e-16 ***
Meg3            4.731e+13  8.077e+05   58571895   &lt;2e-16 ***
Eif4a2         -3.366e+13  7.879e+05  -42721584   &lt;2e-16 ***
Rab10          -3.012e+13  7.727e+05  -38981490   &lt;2e-16 ***
Mllt11          9.241e+13  1.233e+06   74972937   &lt;2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance:  653.13  on 472  degrees of freedom
Residual deviance: 4685.67  on 326  degrees of freedom
AIC: 4979.7

Number of Fisher Scoring iterations: 32</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Make predictions</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>probabilities <span class="ot">&lt;-</span> model <span class="sc">%&gt;%</span> <span class="fu">predict</span>(test.data, <span class="at">type =</span> <span class="st">"response"</span>)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>predicted.classes <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(probabilities <span class="sc">&gt;</span> <span class="fl">0.5</span>, <span class="st">"1"</span>, <span class="st">"0"</span>)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Model accuracy</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(predicted.classes <span class="sc">==</span> test.data<span class="sc">$</span>COTAN_Cl.code)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.6949153</code></pre>
</div>
</div>
</section>
<section id="e13.5-e14.5" class="level2">
<h2 class="anchored" data-anchor-id="e13.5-e14.5">434 e13.5 e14.5</h2>
<hr>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.time</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "2025-12-16 18:58:42 CET"</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.5.2 (2025-10-31)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 22.04.5 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0 
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       
 [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   
 [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          
[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   

time zone: Europe/Rome
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] nnet_7.3-20        COTAN_2.9.4        caret_7.0-1        lattice_0.22-7    
 [5] lubridate_1.9.4    forcats_1.0.0      stringr_1.5.1      dplyr_1.1.4       
 [9] purrr_1.2.0        readr_2.1.5        tibble_3.3.0       tidyverse_2.0.0   
[13] tidyr_1.3.1        parallelDist_0.2.6 data.table_1.17.0  zeallot_0.2.0     
[17] ggplot2_4.0.1     

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.22            splines_4.5.2              
  [3] later_1.4.2                 polyclip_1.10-7            
  [5] hardhat_1.4.1               pROC_1.18.5                
  [7] rpart_4.1.24                fastDummies_1.7.5          
  [9] lifecycle_1.0.4             doParallel_1.0.17          
 [11] processx_3.8.6              globals_0.18.0             
 [13] MASS_7.3-65                 dendextend_1.19.0          
 [15] magrittr_2.0.4              plotly_4.11.0              
 [17] rmarkdown_2.29              yaml_2.3.10                
 [19] httpuv_1.6.16               Seurat_5.2.1               
 [21] sctransform_0.4.2           spam_2.11-1                
 [23] sp_2.2-0                    spatstat.sparse_3.1-0      
 [25] reticulate_1.42.0           cowplot_1.1.3              
 [27] pbapply_1.7-2               RColorBrewer_1.1-3         
 [29] abind_1.4-8                 GenomicRanges_1.60.0       
 [31] Rtsne_0.17                  BiocGenerics_0.54.0        
 [33] coro_1.1.0                  torch_0.14.2               
 [35] ipred_0.9-15                GenomeInfoDbData_1.2.14    
 [37] circlize_0.4.16             lava_1.8.1                 
 [39] IRanges_2.42.0              S4Vectors_0.46.0           
 [41] ggrepel_0.9.6               irlba_2.3.5.1              
 [43] listenv_0.9.1               spatstat.utils_3.1-4       
 [45] goftest_1.2-3               RSpectra_0.16-2            
 [47] spatstat.random_3.4-1       fitdistrplus_1.2-2         
 [49] parallelly_1.45.0           codetools_0.2-20           
 [51] DelayedArray_0.34.1         tidyselect_1.2.1           
 [53] shape_1.4.6.1               UCSC.utils_1.4.0           
 [55] farver_2.1.2                viridis_0.6.5              
 [57] ScaledMatrix_1.16.0         matrixStats_1.5.0          
 [59] stats4_4.5.2                spatstat.explore_3.4-2     
 [61] jsonlite_2.0.0              GetoptLong_1.0.5           
 [63] gghalves_0.1.4              progressr_0.15.1           
 [65] ggridges_0.5.6              survival_3.8-3             
 [67] iterators_1.0.14            foreach_1.5.2              
 [69] tools_4.5.2                 ica_1.0-3                  
 [71] Rcpp_1.0.14                 glue_1.8.0                 
 [73] prodlim_2025.04.28          gridExtra_2.3              
 [75] SparseArray_1.8.0           xfun_0.52                  
 [77] MatrixGenerics_1.20.0       ggthemes_5.1.0             
 [79] GenomeInfoDb_1.44.0         withr_3.0.2                
 [81] fastmap_1.2.0               callr_3.7.6                
 [83] digest_0.6.37               rsvd_1.0.5                 
 [85] timechange_0.3.0            R6_2.6.1                   
 [87] mime_0.13                   colorspace_2.1-1           
 [89] scattermore_1.2             tensor_1.5                 
 [91] spatstat.data_3.1-6         generics_0.1.3             
 [93] recipes_1.3.0               class_7.3-23               
 [95] httr_1.4.7                  htmlwidgets_1.6.4          
 [97] S4Arrays_1.8.0              uwot_0.2.3                 
 [99] ModelMetrics_1.2.2.2        pkgconfig_2.0.3            
[101] gtable_0.3.6                timeDate_4041.110          
[103] ComplexHeatmap_2.24.0       lmtest_0.9-40              
[105] S7_0.2.1                    SingleCellExperiment_1.30.0
[107] XVector_0.48.0              htmltools_0.5.8.1          
[109] dotCall64_1.2               zigg_0.0.2                 
[111] clue_0.3-66                 Biobase_2.68.0             
[113] SeuratObject_5.1.0          scales_1.4.0               
[115] png_0.1-8                   gower_1.0.2                
[117] spatstat.univar_3.1-3       knitr_1.50                 
[119] tzdb_0.5.0                  reshape2_1.4.4             
[121] rjson_0.2.23                nlme_3.1-168               
[123] proxy_0.4-27                zoo_1.8-14                 
[125] GlobalOptions_0.1.2         KernSmooth_2.23-26         
[127] parallel_4.5.2              miniUI_0.1.2               
[129] pillar_1.10.2               grid_4.5.2                 
[131] vctrs_0.6.5                 RANN_2.6.2                 
[133] promises_1.3.2              BiocSingular_1.24.0        
[135] beachmat_2.24.0             xtable_1.8-4               
[137] cluster_2.1.8.1             evaluate_1.0.3             
[139] cli_3.6.5                   compiler_4.5.2             
[141] rlang_1.1.6                 crayon_1.5.3               
[143] future.apply_1.20.0         labeling_0.4.3             
[145] ps_1.9.1                    plyr_1.8.9                 
[147] stringi_1.8.7               deldir_2.0-4               
[149] viridisLite_0.4.2           BiocParallel_1.42.0        
[151] assertthat_0.2.1            lazyeval_0.2.2             
[153] spatstat.geom_3.4-1         Matrix_1.7-4               
[155] RcppHNSW_0.6.0              hms_1.1.3                  
[157] patchwork_1.3.2             bit64_4.6.0-1              
[159] future_1.58.0               shiny_1.11.0               
[161] SummarizedExperiment_1.38.1 ROCR_1.0-11                
[163] Rfast_2.1.5.1               igraph_2.1.4               
[165] RcppParallel_5.1.10         bit_4.6.0                  </code></pre>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/seriph78\.github\.io\/COTAN_Datasets_analysis\/");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>