<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Silvia Galfrè">
<meta name="dcterms.date" content="2024-06-22">

<title>COTAN Datasets analysis - Real UT clusters</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./NonCT_clusters.html">Cluster evaluation</a></li><li class="breadcrumb-item"><a href="./Real_UT_clustersByCotan.html">Real UT clusters with LR</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">COTAN Datasets analysis</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Home</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false">
 <span class="menu-text">Datasets</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false">
 <span class="menu-text">Mouse Cortex Loo 2019</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Mouse_Cortex_E14.5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cleaning and cell clustering</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneCoexAnalysisDataset1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Coexpression analaysis</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false">
 <span class="menu-text">Mouse Cortex Yuzwa 2017</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./CorticalCells_GSM2861511_E135-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">E13.5 Cleaning and cell clustering</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneCoexAnalysisE13.5Yuzwa.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Correlation analysis for E13.5</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./CorticalCells_GSM2861514_E175-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">E17.5 Cleaning and cell clustering</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneCoexAnalysisE17.5Yuzwa.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Correlation analysis for E17.5</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="false">
 <span class="menu-text">Mouse Brain LaManno 2021</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ForebrainDorsal_E135-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">E13.5 Cleaning and cell clustering</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneCoexAnalysisE13.5MouseBrain.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Correlation analysis for E13.5</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ForebrainDorsal_E150-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">E15.0 Cleaning and cell clustering</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneCoexAnalysisE15.0MouseBrain.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Correlation analysis for E15.0</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ForebrainDorsal_E175-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">E17.5 Cleaning and cell clustering</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneCoexAnalysisE17.5MouseBrain.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Correlation analysis for E17.5</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ForebrainLaManno2021ClustersGDI.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Original Clusters GDI evaluation</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false">
 <span class="menu-text">CD14+</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./cd14_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CD14+ Cleaning and cell clustering</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false">
 <span class="menu-text">PBMC form 10X site</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./PBMC1_filtering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cleaning PBMC1</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./PBMC2_filtering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cleaning PBMC2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./PBMC3_filtering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cleaning PBMC3</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./PBMC4_filtering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cleaning PBMC4</span></a>
  </div>
</li>
      </ul>
  </li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" aria-expanded="false">
 <span class="menu-text">Coex score evaluation</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-7" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./generalCoexAssess.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">General Coex Evaluation</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" aria-expanded="false">
 <span class="menu-text">GDI sensitivity</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-8" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./OriginalClustersDistanceEvaluation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Clusters Distance Evaluation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./FindUniformOrigCluster.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cluster Unifomity and GDI score</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./GDI_Increment_From_Mixing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">GDI Increment From Mixing</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-9" aria-expanded="true">
 <span class="menu-text">Cluster evaluation</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-9" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-9" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./NonCT_clusters.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Not UT cluster with LR</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Real_UT_clustersByCotan.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Real UT clusters with LR</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./cd14_clusterizations_comparisons.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CD14+ clusterizations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./clustering_comparison.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PBMCs clustering comparison</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./probability_score.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PBMCs probability score</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-10" aria-expanded="false">
 <span class="menu-text">DEA evaluation</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-10" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-10" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./DEA_Type1Er.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Type I error</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneExpressionAnalysisUniformClusters.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Explorative analysis of gene expression</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./FDR_analisys.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">FDR analysis - define DE genes</span></a>
  </div>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-11" aria-expanded="false">
 <span class="menu-text">FDR and TPR - ROC curves</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-11" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-11" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./FDR_analisysResults3_10.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Results - thresholds true 3% 10%</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./FDR_analisysResults3_10_th04.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Results - thresholds true 3% 10% threshold 0.4</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./FDR_analisysResults5_20.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Results - thresholds true 5% 20%</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./FDR_analisysResults5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Results - thresholds true 5%</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./FDR_analysisResultsLFC.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Results - Log Fold Change</span></a>
  </div>
</li>
      </ul>
  </li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-12" aria-expanded="false">
 <span class="menu-text">Additional code and scripts</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-12" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-12" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./src/scanpyTypeIError.py" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ScamPy Type I Error code</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./src/scanpyFDR.py" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ScamPy FDR code</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scanpy.ipynb" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Example of scanpy clustering and DEA</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./scvi-tools.ipynb" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Example of ScVI clustering and DEA</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#preamble" id="toc-preamble" class="nav-link active" data-scroll-target="#preamble">Preamble</a></li>
  <li><a href="#and-434-e15.0" id="toc-and-434-e15.0" class="nav-link" data-scroll-target="#and-434-e15.0">428 and 434 e15.0</a></li>
  <li><a href="#e13.5-e14.5" id="toc-e13.5-e14.5" class="nav-link" data-scroll-target="#e13.5-e14.5">434 e13.5 e14.5</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Real UT clusters</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Silvia Galfrè </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">June 22, 2024</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<p>Cl432 E13.5 E15.0</p>
<section id="preamble" class="level2">
<h2 class="anchored" data-anchor-id="preamble">Preamble</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(zeallot)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(parallelDist)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(caret)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_bw</span>())</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(COTAN)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(nnet)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">parallelly.fork.enable =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>inDir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"Data/MouseCortexFromLoom/"</span>)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="co">#setLoggingLevel(2)</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="co">#setLoggingFile(file.path(inDir, "MixingClustersGDI_ForebrainDorsal.log"))</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>outDir <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"Data/MouseCortexFromLoom/PureClusters/"</span>)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(outDir)) {</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(outDir)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="and-434-e15.0" class="level2">
<h2 class="anchored" data-anchor-id="and-434-e15.0">428 and 434 e15.0</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>e15<span class="fl">.0</span> <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">paste0</span>(inDir,<span class="st">"e15.0_ForebrainDorsal.cotan.RDS"</span>))</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>cells <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rownames</span>(<span class="fu">getMetadataCells</span>(e15<span class="fl">.0</span>)[<span class="fu">getMetadataCells</span>(e15<span class="fl">.0</span>)[,<span class="st">"Clusters"</span>] <span class="sc">==</span> <span class="dv">428</span>,]),</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>           <span class="fu">rownames</span>(<span class="fu">getMetadataCells</span>(e15<span class="fl">.0</span>)[<span class="fu">getMetadataCells</span>(e15<span class="fl">.0</span>)[,<span class="st">"Clusters"</span>] <span class="sc">==</span> <span class="dv">434</span>,]))</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>cl428_cl434 <span class="ot">&lt;-</span> <span class="fu">dropGenesCells</span>(e15<span class="fl">.0</span>,<span class="at">cells =</span> <span class="fu">getCells</span>(e15<span class="fl">.0</span>)[<span class="sc">!</span><span class="fu">getCells</span>(e15<span class="fl">.0</span>) <span class="sc">%in%</span> cells] )</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>cl428_cl434 <span class="ot">&lt;-</span> <span class="fu">clean</span>(cl428_cl434)</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>cl428_cl434 <span class="ot">&lt;-</span> <span class="fu">proceedToCoex</span>(cl428_cl434)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="co">#saveRDS(cl428_cl434,past0(outDir,"cl428_cl434.cotan.RDS"))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>cl428_cl434GDI <span class="ot">&lt;-</span> <span class="fu">calculateGDI</span>(cl428_cl434)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>subsetGDIcl428_cl434 <span class="ot">&lt;-</span> cl428_cl434GDI[cl428_cl434GDI<span class="sc">$</span>sum.raw.norm <span class="sc">&gt;</span> <span class="dv">7</span>,]</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>top.GDI.genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(subsetGDIcl428_cl434[<span class="fu">order</span>(subsetGDIcl428_cl434<span class="sc">$</span>GDI,<span class="at">decreasing =</span> T),])[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>]</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">GDIPlot</span>(cl428_cl434,<span class="at">genes =</span> <span class="st">""</span>,<span class="at">GDIIn =</span> cl428_cl434GDI)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="Real_UT_clustersByCotan_files/figure-html/unnamed-chunk-2-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">getNormalizedData</span>(cl428_cl434)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> data<span class="sc">/</span><span class="fu">sum</span>(data[,<span class="dv">1</span>])</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co">#data &lt;- data[!rowSums(as.matrix(data)) &lt; 1,]</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">log</span>(data<span class="sc">*</span><span class="dv">10000</span><span class="sc">+</span><span class="dv">1</span>)</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>row_stdev <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">1</span>, sd, <span class="at">na.rm=</span><span class="cn">TRUE</span>)</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>row_stdev <span class="ot">&lt;-</span> row_stdev[<span class="fu">order</span>(row_stdev,<span class="at">decreasing =</span> T)]</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>genes.to.keep <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">names</span>(row_stdev[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>]),top.GDI.genes)</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>data.small <span class="ot">&lt;-</span> data[<span class="fu">rownames</span>(data) <span class="sc">%in%</span> genes.to.keep,]</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="co">#data &lt;- t(as.matrix(data))</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>data.small <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">as.matrix</span>(data.small))</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>COTAN_Cl.code <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">getClusterizationData</span>(cl428_cl434,<span class="at">clName =</span> <span class="st">"original.clusters"</span>)[[<span class="dv">1</span>]])</span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>COTAN_Cl.code <span class="ot">&lt;-</span> COTAN_Cl.code <span class="sc">-</span><span class="dv">1</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>data.small <span class="ot">&lt;-</span> <span class="fu">cbind</span>(data.small,COTAN_Cl.code)</span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>data.small <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(data.small)</span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Split the data into training and test set</span></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>training.samples <span class="ot">&lt;-</span> data.small[,<span class="st">"COTAN_Cl.code"</span>] <span class="sc">%&gt;%</span> </span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">createDataPartition</span>(<span class="at">p =</span> <span class="fl">0.8</span>, <span class="at">list =</span> <span class="cn">FALSE</span>)</span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a>train.data  <span class="ot">&lt;-</span> data.small[training.samples, ]</span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>test.data <span class="ot">&lt;-</span> data.small[<span class="sc">-</span>training.samples, ]</span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(train.data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                              Hs3st1   Abracl   Cdkn1c   Cited2    Mllt3
10X74_4_A_1:GTAACGTGCCATAGx 2.131073 2.131073 0.000000 1.550102 1.550102
10X73_3_A_1:ACGCCGGATCGTAGx 1.615546 2.204009 0.000000 2.840395 2.204009
10X74_4_A_1:GATTCGGAGGTGAGx 0.000000 2.481727 1.319213 1.868868 1.319213
10X74_4_A_1:CTGAGCCTGGTAAAx 2.018835 2.852085 0.000000 1.450389 2.379020
10X74_4_A_1:CCTAGAGAGTCACAx 2.634119 1.669657 0.000000 3.115812 0.000000
10X74_4_A_1:CTAGGATGCTTGCCx 0.000000 3.728844 0.000000 2.875281 1.220847
                               Mfap4   Cxcl12 Cdc42ep3    Fmnl2  Phf20l1
10X74_4_A_1:GTAACGTGCCATAGx 1.550102 1.550102 0.000000 0.000000 0.000000
10X73_3_A_1:ACGCCGGATCGTAGx 1.615546 0.000000 1.615546 2.571994 2.571994
10X74_4_A_1:GATTCGGAGGTGAGx 3.004857 1.868868 0.000000 0.000000 1.319213
10X74_4_A_1:CTGAGCCTGGTAAAx 2.379020 0.000000 2.018835 0.000000 2.018835
10X74_4_A_1:CCTAGAGAGTCACAx 1.669657 1.669657 0.000000 0.000000 1.669657
10X74_4_A_1:CTAGGATGCTTGCCx 2.875281 1.754424 0.000000 0.000000 0.000000
                               Ptprk     Tnik  Gm29260    Sfrp1   Pantr1
10X74_4_A_1:GTAACGTGCCATAGx 1.550102 1.550102 0.000000 1.550102 2.496164
10X73_3_A_1:ACGCCGGATCGTAGx 2.204009 2.204009 1.615546 1.615546 1.615546
10X74_4_A_1:GATTCGGAGGTGAGx 1.319213 1.868868 1.868868 1.868868 2.481727
10X74_4_A_1:CTGAGCCTGGTAAAx 1.450389 0.000000 2.018835 0.000000 2.379020
10X74_4_A_1:CCTAGAGAGTCACAx 1.669657 0.000000 0.000000 1.669657 0.000000
10X74_4_A_1:CTAGGATGCTTGCCx 0.000000 0.000000 1.220847 1.754424 2.561118
                               Magi2  Tmem108   Frmd4b   Plppr1     Nfia
10X74_4_A_1:GTAACGTGCCATAGx 0.000000 0.000000 0.000000 0.000000 0.000000
10X73_3_A_1:ACGCCGGATCGTAGx 1.615546 0.000000 1.615546 2.840395 2.204009
10X74_4_A_1:GATTCGGAGGTGAGx 0.000000 0.000000 0.000000 2.221530 1.868868
10X74_4_A_1:CTGAGCCTGGTAAAx 0.000000 1.450389 2.379020 2.018835 2.018835
10X74_4_A_1:CCTAGAGAGTCACAx 0.000000 0.000000 0.000000 1.669657 1.669657
10X74_4_A_1:CTAGGATGCTTGCCx 2.357095 1.220847 1.754424 1.220847 2.730486
                              Igfbp2    Fabp7 Ctnnd2      Ptn   Frmd4a   Nkain2
10X74_4_A_1:GTAACGTGCCATAGx 2.131073 0.000000      0 2.131073 3.147239 1.550102
10X73_3_A_1:ACGCCGGATCGTAGx 0.000000 1.615546      0 1.615546 1.615546 3.051789
10X74_4_A_1:GATTCGGAGGTGAGx 2.858930 1.319213      0 0.000000 0.000000 1.868868
10X74_4_A_1:CTGAGCCTGGTAAAx 2.018835 1.450389      0 2.643269 1.450389 2.643269
10X74_4_A_1:CCTAGAGAGTCACAx 0.000000 1.669657      0 2.263916 0.000000 0.000000
10X74_4_A_1:CTAGGATGCTTGCCx 2.357095 1.754424      0 0.000000 0.000000 1.220847
                              Sorbs2     Hes6      Dcc  Neurog2    Tcf12
10X74_4_A_1:GTAACGTGCCATAGx 2.496164 0.000000 1.550102 0.000000 0.000000
10X73_3_A_1:ACGCCGGATCGTAGx 2.571994 1.615546 0.000000 0.000000 2.571994
10X74_4_A_1:GATTCGGAGGTGAGx 1.868868 1.868868 1.868868 2.688010 1.868868
10X74_4_A_1:CTGAGCCTGGTAAAx 2.018835 0.000000 2.852085 0.000000 2.018835
10X74_4_A_1:CCTAGAGAGTCACAx 1.669657 1.669657 0.000000 0.000000 3.115812
10X74_4_A_1:CTAGGATGCTTGCCx 2.100490 0.000000 1.220847 1.220847 2.100490
                             Cbfa2t2    Ccnd2   Elavl4    Clvs1     Ank3
10X74_4_A_1:GTAACGTGCCATAGx 0.000000 0.000000 1.550102 0.000000 1.550102
10X73_3_A_1:ACGCCGGATCGTAGx 1.615546 1.615546 0.000000 0.000000 1.615546
10X74_4_A_1:GATTCGGAGGTGAGx 2.481727 2.858930 1.319213 0.000000 0.000000
10X74_4_A_1:CTGAGCCTGGTAAAx 2.018835 0.000000 2.018835 0.000000 2.852085
10X74_4_A_1:CCTAGAGAGTCACAx 0.000000 2.263916 2.903693 2.634119 1.669657
10X74_4_A_1:CTAGGATGCTTGCCx 1.220847 2.100490 2.100490 1.220847 1.754424
                               Igsf8     Map2     Lhx2   Nkain3  Neurod2
10X74_4_A_1:GTAACGTGCCATAGx 0.000000 0.000000 2.131073 1.550102 2.496164
10X73_3_A_1:ACGCCGGATCGTAGx 1.615546 2.204009 1.615546 1.615546 1.615546
10X74_4_A_1:GATTCGGAGGTGAGx 1.319213 1.868868 2.221530 2.688010 1.868868
10X74_4_A_1:CTGAGCCTGGTAAAx 1.450389 0.000000 2.018835 3.171927 3.024739
10X74_4_A_1:CCTAGAGAGTCACAx 1.669657 2.903693 2.903693 0.000000 2.263916
10X74_4_A_1:CTAGGATGCTTGCCx 1.754424 1.220847 2.730486 0.000000 2.561118
                               Sstr2    Eomes C130071C03Rik     Emx1  Cttnbp2
10X74_4_A_1:GTAACGTGCCATAGx 2.763031 0.000000      2.131073 2.131073 0.000000
10X73_3_A_1:ACGCCGGATCGTAGx 2.204009 0.000000      2.204009 1.615546 3.051789
10X74_4_A_1:GATTCGGAGGTGAGx 2.481727 1.868868      0.000000 1.319213 2.221530
10X74_4_A_1:CTGAGCCTGGTAAAx 2.643269 0.000000      2.643269 1.450389 0.000000
10X74_4_A_1:CCTAGAGAGTCACAx 2.263916 2.263916      1.669657 0.000000 3.115812
10X74_4_A_1:CTAGGATGCTTGCCx 2.875281 0.000000      1.220847 2.100490 2.875281
                                Nfix    Epha4   Plxna2   Plxna4     Aff3
10X74_4_A_1:GTAACGTGCCATAGx 2.131073 0.000000 1.550102 1.550102 1.550102
10X73_3_A_1:ACGCCGGATCGTAGx 3.721016 0.000000 1.615546 0.000000 2.571994
10X74_4_A_1:GATTCGGAGGTGAGx 2.688010 1.319213 2.221530 0.000000 0.000000
10X74_4_A_1:CTGAGCCTGGTAAAx 3.024739 1.450389 2.643269 0.000000 0.000000
10X74_4_A_1:CCTAGAGAGTCACAx 2.903693 1.669657 1.669657 0.000000 0.000000
10X74_4_A_1:CTAGGATGCTTGCCx 3.467944 1.220847 0.000000 0.000000 0.000000
                                Nrp1   Nedd4l    Tiam2  Ppp2r2b   Bcl11a
10X74_4_A_1:GTAACGTGCCATAGx 2.131073 1.550102 1.550102 2.763031 2.131073
10X73_3_A_1:ACGCCGGATCGTAGx 2.204009 0.000000 2.204009 2.204009 1.615546
10X74_4_A_1:GATTCGGAGGTGAGx 0.000000 0.000000 2.221530 1.319213 1.868868
10X74_4_A_1:CTGAGCCTGGTAAAx 2.018835 0.000000 1.450389 2.852085 2.018835
10X74_4_A_1:CCTAGAGAGTCACAx 0.000000 1.669657 0.000000 2.263916 0.000000
10X74_4_A_1:CTAGGATGCTTGCCx 0.000000 1.220847 2.100490 2.100490 2.561118
                                Zeb2    Tenm2    Gria2    Gpm6a    Ptprd
10X74_4_A_1:GTAACGTGCCATAGx 1.550102 0.000000 2.496164 1.550102 1.550102
10X73_3_A_1:ACGCCGGATCGTAGx 0.000000 0.000000 2.204009 2.840395 0.000000
10X74_4_A_1:GATTCGGAGGTGAGx 1.868868 0.000000 1.319213 1.319213 0.000000
10X74_4_A_1:CTGAGCCTGGTAAAx 0.000000 0.000000 2.379020 1.450389 2.379020
10X74_4_A_1:CCTAGAGAGTCACAx 1.669657 0.000000 1.669657 0.000000 0.000000
10X74_4_A_1:CTAGGATGCTTGCCx 2.357095 1.220847 0.000000 0.000000 0.000000
                                Sox5   Mpped2   Bcl11b    Epha5     Nol4
10X74_4_A_1:GTAACGTGCCATAGx 2.763031 2.496164 1.550102 2.496164 0.000000
10X73_3_A_1:ACGCCGGATCGTAGx 1.615546 2.204009 0.000000 2.571994 2.204009
10X74_4_A_1:GATTCGGAGGTGAGx 1.319213 2.858930 0.000000 2.221530 0.000000
10X74_4_A_1:CTGAGCCTGGTAAAx 2.018835 1.450389 0.000000 2.018835 2.852085
10X74_4_A_1:CCTAGAGAGTCACAx 2.263916 0.000000 0.000000 2.634119 2.903693
10X74_4_A_1:CTAGGATGCTTGCCx 0.000000 1.754424 0.000000 2.357095 0.000000
                               Grik2     Pcp4   Ccser1    Kcnb2    Kcnh7
10X74_4_A_1:GTAACGTGCCATAGx 1.550102 0.000000 3.538277 2.131073 0.000000
10X73_3_A_1:ACGCCGGATCGTAGx 0.000000 3.051789 2.840395 0.000000 0.000000
10X74_4_A_1:GATTCGGAGGTGAGx 1.319213 2.688010 0.000000 1.319213 1.319213
10X74_4_A_1:CTGAGCCTGGTAAAx 0.000000 0.000000 3.024739 0.000000 1.450389
10X74_4_A_1:CCTAGAGAGTCACAx 0.000000 2.903693 2.263916 0.000000 1.669657
10X74_4_A_1:CTAGGATGCTTGCCx 1.220847 1.754424 2.357095 1.220847 2.357095
                                 Pam    Negr1    Myt1l      Id2    Foxp2
10X74_4_A_1:GTAACGTGCCATAGx 0.000000 1.550102 3.424120 2.973475 1.550102
10X73_3_A_1:ACGCCGGATCGTAGx 2.571994 2.571994 2.204009 2.204009 1.615546
10X74_4_A_1:GATTCGGAGGTGAGx 1.868868 2.481727 0.000000 0.000000 0.000000
10X74_4_A_1:CTGAGCCTGGTAAAx 1.450389 0.000000 2.379020 0.000000 2.379020
10X74_4_A_1:CCTAGAGAGTCACAx 2.634119 3.115812 1.669657 2.903693 0.000000
10X74_4_A_1:CTAGGATGCTTGCCx 1.220847 0.000000 2.100490 2.561118 0.000000
                                Gpc6    Pcdh7   Rbfox1    Robo2     Rprm
10X74_4_A_1:GTAACGTGCCATAGx 2.131073 2.131073 2.763031 3.295232 0.000000
10X73_3_A_1:ACGCCGGATCGTAGx 2.204009 3.051789 3.374663 3.618342 0.000000
10X74_4_A_1:GATTCGGAGGTGAGx 0.000000 1.319213 1.319213 3.245101 1.868868
10X74_4_A_1:CTGAGCCTGGTAAAx 0.000000 0.000000 3.024739 2.643269 0.000000
10X74_4_A_1:CCTAGAGAGTCACAx 0.000000 2.903693 1.669657 2.634119 0.000000
10X74_4_A_1:CTAGGATGCTTGCCx 0.000000 2.100490 0.000000 3.113983 1.220847
                              Rbfox3    Stmn3     Nrn1     Nrg1     Cdh8
10X74_4_A_1:GTAACGTGCCATAGx 1.550102 1.550102 0.000000 2.973475 2.131073
10X73_3_A_1:ACGCCGGATCGTAGx 0.000000 0.000000 0.000000 2.840395 3.226201
10X74_4_A_1:GATTCGGAGGTGAGx 1.868868 0.000000 1.868868 2.688010 2.688010
10X74_4_A_1:CTGAGCCTGGTAAAx 2.379020 2.379020 2.852085 3.300205 2.379020
10X74_4_A_1:CCTAGAGAGTCACAx 1.669657 1.669657 1.669657 2.903693 2.263916
10X74_4_A_1:CTAGGATGCTTGCCx 1.220847 2.357095 1.754424 3.113983 2.357095
                            Cacna2d1    Rpl26    Rack1     Top1    Rpl12
10X74_4_A_1:GTAACGTGCCATAGx 0.000000 2.763031 0.000000 2.496164 1.550102
10X73_3_A_1:ACGCCGGATCGTAGx 2.204009 3.226201 0.000000 1.615546 0.000000
10X74_4_A_1:GATTCGGAGGTGAGx 2.481727 2.858930 2.221530 2.221530 1.868868
10X74_4_A_1:CTGAGCCTGGTAAAx 2.018835 2.852085 1.450389 1.450389 0.000000
10X74_4_A_1:CCTAGAGAGTCACAx 1.669657 2.634119 2.263916 1.669657 1.669657
10X74_4_A_1:CTAGGATGCTTGCCx 1.220847 3.214891 2.100490 2.100490 2.357095
                               Rplp1    Rpl22     Eif5   Rpl36a    Rps21
10X74_4_A_1:GTAACGTGCCATAGx 2.131073 0.000000 2.973475 2.496164 1.550102
10X73_3_A_1:ACGCCGGATCGTAGx 2.571994 0.000000 0.000000 0.000000 0.000000
10X74_4_A_1:GATTCGGAGGTGAGx 2.481727 1.319213 1.868868 0.000000 2.688010
10X74_4_A_1:CTGAGCCTGGTAAAx 1.450389 2.018835 2.018835 3.024739 1.450389
10X74_4_A_1:CCTAGAGAGTCACAx 2.634119 0.000000 0.000000 2.634119 1.669657
10X74_4_A_1:CTAGGATGCTTGCCx 2.100490 1.754424 1.220847 2.100490 2.357095
                              Eif4a1     Ppia     Sub1      Mdk    Pde4d
10X74_4_A_1:GTAACGTGCCATAGx 1.550102 2.763031 2.131073 0.000000 1.550102
10X73_3_A_1:ACGCCGGATCGTAGx 2.204009 2.204009 0.000000 2.571994 2.571994
10X74_4_A_1:GATTCGGAGGTGAGx 1.868868 1.319213 1.868868 3.004857 0.000000
10X74_4_A_1:CTGAGCCTGGTAAAx 2.852085 1.450389 1.450389 3.693302 0.000000
10X74_4_A_1:CCTAGAGAGTCACAx 2.263916 0.000000 1.669657 3.115812 1.669657
10X74_4_A_1:CTAGGATGCTTGCCx 1.754424 0.000000 1.220847 2.875281 2.875281
                               Rps28    Rpl15    Baz2b     Rere   Rpl35a
10X74_4_A_1:GTAACGTGCCATAGx 1.550102 2.496164 1.550102 1.550102 2.763031
10X73_3_A_1:ACGCCGGATCGTAGx 1.615546 0.000000 1.615546 3.051789 1.615546
10X74_4_A_1:GATTCGGAGGTGAGx 1.319213 2.481727 1.868868 0.000000 2.481727
10X74_4_A_1:CTGAGCCTGGTAAAx 2.018835 2.379020 1.450389 3.300205 2.018835
10X74_4_A_1:CCTAGAGAGTCACAx 0.000000 2.634119 0.000000 1.669657 0.000000
10X74_4_A_1:CTAGGATGCTTGCCx 1.220847 2.561118 0.000000 1.754424 1.754424
                               Cox8a   Nt5dc2    Ttc28      Ckb     Calr
10X74_4_A_1:GTAACGTGCCATAGx 2.131073 0.000000 2.131073 1.550102 0.000000
10X73_3_A_1:ACGCCGGATCGTAGx 2.204009 0.000000 2.571994 0.000000 0.000000
10X74_4_A_1:GATTCGGAGGTGAGx 1.319213 2.858930 2.688010 1.319213 0.000000
10X74_4_A_1:CTGAGCCTGGTAAAx 1.450389 2.018835 3.171927 0.000000 2.018835
10X74_4_A_1:CCTAGAGAGTCACAx 0.000000 0.000000 2.263916 0.000000 1.669657
10X74_4_A_1:CTAGGATGCTTGCCx 1.220847 2.100490 1.754424 1.220847 0.000000
                               Arpc5    Basp1    Map1b     Xist   Ctnna2
10X74_4_A_1:GTAACGTGCCATAGx 2.131073 2.496164 2.496164 0.000000 2.496164
10X73_3_A_1:ACGCCGGATCGTAGx 1.615546 2.571994 1.615546 2.571994 3.374663
10X74_4_A_1:GATTCGGAGGTGAGx 1.319213 2.481727 0.000000 0.000000 1.868868
10X74_4_A_1:CTGAGCCTGGTAAAx 0.000000 1.450389 2.018835 2.643269 2.018835
10X74_4_A_1:CCTAGAGAGTCACAx 1.669657 0.000000 1.669657 3.683694 0.000000
10X74_4_A_1:CTAGGATGCTTGCCx 2.100490 1.754424 0.000000 2.357095 1.754424
                               Fabp5   Zbtb20    Eif1b     Nsg1     Meg3
10X74_4_A_1:GTAACGTGCCATAGx 0.000000 1.550102 1.550102 1.550102 0.000000
10X73_3_A_1:ACGCCGGATCGTAGx 1.615546 2.204009 2.571994 1.615546 0.000000
10X74_4_A_1:GATTCGGAGGTGAGx 1.868868 2.481727 1.319213 1.868868 0.000000
10X74_4_A_1:CTGAGCCTGGTAAAx 2.018835 1.450389 2.643269 2.643269 2.018835
10X74_4_A_1:CCTAGAGAGTCACAx 0.000000 2.263916 1.669657 2.634119 1.669657
10X74_4_A_1:CTAGGATGCTTGCCx 2.357095 1.220847 2.100490 1.754424 1.220847
                               Rsrp1   Mllt11    Ctxn1   Dpysl3 COTAN_Cl.code
10X74_4_A_1:GTAACGTGCCATAGx 2.496164 2.131073 1.550102 1.550102             0
10X73_3_A_1:ACGCCGGATCGTAGx 1.615546 0.000000 2.840395 1.615546             0
10X74_4_A_1:GATTCGGAGGTGAGx 1.868868 1.319213 1.319213 0.000000             0
10X74_4_A_1:CTGAGCCTGGTAAAx 2.018835 2.379020 0.000000 2.379020             0
10X74_4_A_1:CCTAGAGAGTCACAx 1.669657 1.669657 1.669657 0.000000             0
10X74_4_A_1:CTAGGATGCTTGCCx 1.220847 2.357095 2.561118 0.000000             0</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Fit the model</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="fu">glm</span>( COTAN_Cl.code <span class="sc">~</span>., <span class="at">data =</span> train.data, <span class="at">family =</span> binomial,<span class="at">control =</span> <span class="fu">list</span>(<span class="at">maxit =</span> <span class="dv">50</span>))</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize the model</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(model)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
glm(formula = COTAN_Cl.code ~ ., family = binomial, data = train.data, 
    control = list(maxit = 50))

Coefficients:
               Estimate Std. Error z value Pr(&gt;|z|)    
(Intercept)   -4.427802   3.686632  -1.201 0.229735    
Hs3st1        -0.045329   0.273332  -0.166 0.868285    
Abracl        -0.021102   0.287762  -0.073 0.941542    
Cdkn1c         1.282281   0.317326   4.041 5.32e-05 ***
Cited2        -0.095750   0.335083  -0.286 0.775069    
Mllt3         -0.017451   0.291300  -0.060 0.952231    
Mfap4         -0.328338   0.301103  -1.090 0.275515    
Cxcl12         0.374555   0.301754   1.241 0.214510    
Cdc42ep3       0.948126   0.344243   2.754 0.005883 ** 
Fmnl2          0.761696   0.280651   2.714 0.006647 ** 
Phf20l1       -0.049548   0.295853  -0.167 0.866995    
Ptprk         -0.005845   0.310432  -0.019 0.984979    
Tnik           0.068744   0.288077   0.239 0.811392    
Gm29260       -2.170967   0.457504  -4.745 2.08e-06 ***
Sfrp1         -0.320217   0.322417  -0.993 0.320623    
Pantr1        -0.701726   0.342261  -2.050 0.040339 *  
Magi2          0.063704   0.269791   0.236 0.813336    
Tmem108        0.013377   0.296197   0.045 0.963977    
Frmd4b        -0.452528   0.296115  -1.528 0.126459    
Plppr1         0.173573   0.280712   0.618 0.536357    
Nfia           0.193750   0.287645   0.674 0.500583    
Igfbp2        -0.113178   0.335945  -0.337 0.736197    
Fabp7         -0.131296   0.297122  -0.442 0.658567    
Ctnnd2         0.322188   0.280410   1.149 0.250561    
Ptn           -0.594166   0.313001  -1.898 0.057658 .  
Frmd4a        -0.277474   0.255226  -1.087 0.276961    
Nkain2         0.104150   0.293746   0.355 0.722920    
Sorbs2         0.663126   0.298806   2.219 0.026469 *  
Hes6           0.855882   0.323864   2.643 0.008224 ** 
Dcc           -0.505708   0.306444  -1.650 0.098893 .  
Neurog2        0.261606   0.305102   0.857 0.391202    
Tcf12          0.038644   0.274525   0.141 0.888053    
Cbfa2t2       -0.433520   0.268387  -1.615 0.106250    
Ccnd2          0.179213   0.274252   0.653 0.513459    
Elavl4         1.215423   0.399152   3.045 0.002327 ** 
Clvs1          0.089130   0.282731   0.315 0.752575    
Ank3           0.179434   0.282506   0.635 0.525330    
Igsf8          0.129736   0.275953   0.470 0.638255    
Map2          -0.181323   0.310183  -0.585 0.558839    
Lhx2           0.402542   0.349519   1.152 0.249442    
Nkain3        -0.044385   0.277034  -0.160 0.872710    
Neurod2        0.105776   0.378601   0.279 0.779948    
Sstr2         -0.362034   0.300426  -1.205 0.228176    
Eomes          0.772645   0.349417   2.211 0.027019 *  
C130071C03Rik -0.122480   0.263918  -0.464 0.642587    
Emx1          -0.593572   0.267820  -2.216 0.026670 *  
Cttnbp2        0.273984   0.318743   0.860 0.390023    
Nfix           0.517842   0.348689   1.485 0.137514    
Epha4         -0.638904   0.321235  -1.989 0.046712 *  
Plxna2        -0.236110   0.258832  -0.912 0.361657    
Plxna4         0.925213   0.354572   2.609 0.009071 ** 
Aff3          -0.013714   0.295144  -0.046 0.962940    
Nrp1           0.232902   0.272307   0.855 0.392389    
Nedd4l        -0.303610   0.310862  -0.977 0.328732    
Tiam2          0.134778   0.290641   0.464 0.642843    
Ppp2r2b        0.016351   0.367928   0.044 0.964553    
Bcl11a         0.337470   0.274048   1.231 0.218163    
Zeb2          -1.599521   0.393935  -4.060 4.90e-05 ***
Tenm2          0.468738   0.303563   1.544 0.122559    
Gria2          0.149754   0.312789   0.479 0.632103    
Gpm6a          0.119547   0.268907   0.445 0.656633    
Ptprd         -0.507448   0.323955  -1.566 0.117251    
Sox5          -0.256282   0.298164  -0.860 0.390046    
Mpped2         0.103093   0.283960   0.363 0.716564    
Bcl11b        -0.280065   0.348847  -0.803 0.422072    
Epha5          0.749204   0.349295   2.145 0.031961 *  
Nol4           0.098607   0.302192   0.326 0.744194    
Grik2         -0.008390   0.310334  -0.027 0.978432    
Pcp4           0.054691   0.254859   0.215 0.830084    
Ccser1         0.280685   0.303408   0.925 0.354909    
Kcnb2         -0.585578   0.278485  -2.103 0.035489 *  
Kcnh7         -0.492670   0.281629  -1.749 0.080229 .  
Pam            0.142564   0.301995   0.472 0.636874    
Negr1         -0.716179   0.285728  -2.507 0.012193 *  
Myt1l          0.163594   0.292845   0.559 0.576408    
Id2           -0.668509   0.288152  -2.320 0.020342 *  
Foxp2          0.725433   0.271382   2.673 0.007515 ** 
Gpc6           0.069695   0.314905   0.221 0.824843    
Pcdh7          0.814931   0.291051   2.800 0.005111 ** 
Rbfox1        -0.459985   0.294818  -1.560 0.118705    
Robo2          0.144083   0.316589   0.455 0.649030    
Rprm           0.958384   0.297789   3.218 0.001289 ** 
Rbfox3        -0.123098   0.304439  -0.404 0.685960    
Stmn3         -0.779303   0.310707  -2.508 0.012136 *  
Nrn1          -0.293944   0.307378  -0.956 0.338923    
Nrg1          -0.180726   0.303194  -0.596 0.551126    
Cdh8          -0.830419   0.303920  -2.732 0.006288 ** 
Cacna2d1       0.524141   0.335929   1.560 0.118695    
Rpl26         -1.285495   0.369625  -3.478 0.000505 ***
Rack1          0.141107   0.303411   0.465 0.641881    
Top1          -0.888447   0.379757  -2.340 0.019309 *  
Rpl12          0.118293   0.285560   0.414 0.678692    
Rplp1         -0.649347   0.402071  -1.615 0.106309    
Rpl22          0.098887   0.318478   0.310 0.756181    
Eif5          -0.621647   0.305526  -2.035 0.041884 *  
Rpl36a         0.797852   0.333875   2.390 0.016863 *  
Rps21         -0.480747   0.352633  -1.363 0.172786    
Eif4a1         0.665213   0.340406   1.954 0.050681 .  
Ppia          -0.144535   0.305864  -0.473 0.636538    
Sub1           0.308813   0.289513   1.067 0.286124    
Mdk           -0.266772   0.334220  -0.798 0.424759    
Pde4d          1.286486   0.347047   3.707 0.000210 ***
Rps28          0.146676   0.346989   0.423 0.672507    
Rpl15          1.194097   0.386632   3.088 0.002012 ** 
Baz2b          0.214207   0.284718   0.752 0.451843    
Rere          -0.880721   0.324004  -2.718 0.006563 ** 
Rpl35a        -0.465717   0.348718  -1.336 0.181710    
Cox8a          0.157332   0.300227   0.524 0.600249    
Nt5dc2        -0.265734   0.280191  -0.948 0.342923    
Ttc28         -0.550163   0.304121  -1.809 0.070447 .  
Ckb            0.294038   0.306467   0.959 0.337334    
Calr           0.154004   0.305910   0.503 0.614661    
Arpc5         -0.142432   0.273624  -0.521 0.602688    
Basp1          0.160793   0.265227   0.606 0.544350    
Map1b         -0.469132   0.295153  -1.589 0.111957    
Xist          -0.207244   0.247978  -0.836 0.403303    
Ctnna2         0.164829   0.269426   0.612 0.540685    
Fabp5          1.215383   0.345384   3.519 0.000433 ***
Zbtb20        -0.371833   0.309890  -1.200 0.230183    
Eif1b         -0.161688   0.291010  -0.556 0.578477    
Nsg1          -0.040988   0.323469  -0.127 0.899166    
Meg3          -0.051099   0.265693  -0.192 0.847489    
Rsrp1          0.536480   0.291008   1.844 0.065253 .  
Mllt11         0.482180   0.371390   1.298 0.194181    
Ctxn1          0.516132   0.313897   1.644 0.100120    
Dpysl3         0.599812   0.314290   1.908 0.056331 .  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 653.13  on 472  degrees of freedom
Residual deviance: 217.74  on 347  degrees of freedom
AIC: 469.74

Number of Fisher Scoring iterations: 8</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Make predictions</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>probabilities <span class="ot">&lt;-</span> model <span class="sc">%&gt;%</span> <span class="fu">predict</span>(test.data, <span class="at">type =</span> <span class="st">"response"</span>)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>predicted.classes <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(probabilities <span class="sc">&gt;</span> <span class="fl">0.5</span>, <span class="st">"1"</span>, <span class="st">"0"</span>)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Model accuracy</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(predicted.classes <span class="sc">==</span> test.data<span class="sc">$</span>COTAN_Cl.code)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.6949153</code></pre>
</div>
</div>
</section>
<section id="e13.5-e14.5" class="level2">
<h2 class="anchored" data-anchor-id="e13.5-e14.5">434 e13.5 e14.5</h2>
<hr>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.time</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "2024-06-22 16:00:00 CEST"</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 20.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       
 [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   
 [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          
[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   

time zone: Europe/Rome
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] torch_0.13.0       nnet_7.3-19        COTAN_2.5.4        caret_6.0-94      
 [5] lattice_0.22-6     lubridate_1.9.3    forcats_1.0.0      stringr_1.5.1     
 [9] dplyr_1.1.4        purrr_1.0.2        readr_2.1.5        tibble_3.2.1      
[13] tidyverse_2.0.0    tidyr_1.3.1        parallelDist_0.2.6 data.table_1.15.4 
[17] zeallot_0.1.0      ggplot2_3.5.1     

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.22          splines_4.4.0            
  [3] later_1.3.2               polyclip_1.10-6          
  [5] hardhat_1.3.1             pROC_1.18.5              
  [7] rpart_4.1.23              fastDummies_1.7.3        
  [9] lifecycle_1.0.4           doParallel_1.0.17        
 [11] processx_3.8.4            globals_0.16.3           
 [13] MASS_7.3-60.2             dendextend_1.17.1        
 [15] magrittr_2.0.3            plotly_4.10.4            
 [17] rmarkdown_2.27            yaml_2.3.8               
 [19] httpuv_1.6.15             Seurat_5.1.0             
 [21] sctransform_0.4.1         askpass_1.2.0            
 [23] spam_2.10-0               spatstat.sparse_3.0-3    
 [25] sp_2.1-4                  reticulate_1.37.0        
 [27] cowplot_1.1.3             pbapply_1.7-2            
 [29] RColorBrewer_1.1-3        abind_1.4-5              
 [31] zlibbioc_1.50.0           Rtsne_0.17               
 [33] BiocGenerics_0.50.0       coro_1.0.4               
 [35] ipred_0.9-14              circlize_0.4.16          
 [37] lava_1.8.0                IRanges_2.38.0           
 [39] S4Vectors_0.42.0          ggrepel_0.9.5            
 [41] irlba_2.3.5.1             spatstat.utils_3.0-4     
 [43] listenv_0.9.1             umap_0.2.10.0            
 [45] goftest_1.2-3             RSpectra_0.16-1          
 [47] spatstat.random_3.2-3     dqrng_0.4.0              
 [49] fitdistrplus_1.1-11       parallelly_1.37.1        
 [51] DelayedMatrixStats_1.26.0 leiden_0.4.3.1           
 [53] codetools_0.2-20          DelayedArray_0.30.1      
 [55] tidyselect_1.2.1          shape_1.4.6.1            
 [57] farver_2.1.2              viridis_0.6.5            
 [59] ScaledMatrix_1.12.0       spatstat.explore_3.2-7   
 [61] matrixStats_1.3.0         stats4_4.4.0             
 [63] jsonlite_1.8.8            GetoptLong_1.0.5         
 [65] progressr_0.14.0          ggridges_0.5.6           
 [67] survival_3.6-4            iterators_1.0.14         
 [69] foreach_1.5.2             tools_4.4.0              
 [71] ica_1.0-3                 Rcpp_1.0.12              
 [73] glue_1.7.0                gridExtra_2.3            
 [75] prodlim_2023.08.28        SparseArray_1.4.5        
 [77] xfun_0.44                 MatrixGenerics_1.16.0    
 [79] ggthemes_5.1.0            withr_3.0.0              
 [81] fastmap_1.2.0             fansi_1.0.6              
 [83] openssl_2.2.0             callr_3.7.6              
 [85] digest_0.6.35             rsvd_1.0.5               
 [87] timechange_0.3.0          R6_2.5.1                 
 [89] mime_0.12                 colorspace_2.1-0         
 [91] scattermore_1.2           tensor_1.5               
 [93] spatstat.data_3.0-4       utf8_1.2.4               
 [95] generics_0.1.3            recipes_1.0.10           
 [97] class_7.3-22              httr_1.4.7               
 [99] htmlwidgets_1.6.4         S4Arrays_1.4.1           
[101] uwot_0.2.2                ModelMetrics_1.2.2.2     
[103] pkgconfig_2.0.3           gtable_0.3.5             
[105] timeDate_4032.109         ComplexHeatmap_2.20.0    
[107] lmtest_0.9-40             XVector_0.44.0           
[109] htmltools_0.5.8.1         dotCall64_1.1-1          
[111] clue_0.3-65               SeuratObject_5.0.2       
[113] scales_1.3.0              png_0.1-8                
[115] gower_1.0.1               knitr_1.47               
[117] rstudioapi_0.16.0         tzdb_0.4.0               
[119] reshape2_1.4.4            rjson_0.2.21             
[121] nlme_3.1-164              zoo_1.8-12               
[123] GlobalOptions_0.1.2       KernSmooth_2.23-24       
[125] parallel_4.4.0            miniUI_0.1.1.1           
[127] RcppZiggurat_0.1.6        pillar_1.9.0             
[129] grid_4.4.0                vctrs_0.6.5              
[131] RANN_2.6.1                promises_1.3.0           
[133] BiocSingular_1.20.0       beachmat_2.20.0          
[135] xtable_1.8-4              cluster_2.1.6            
[137] evaluate_0.23             cli_3.6.2                
[139] compiler_4.4.0            rlang_1.1.3              
[141] crayon_1.5.2              future.apply_1.11.2      
[143] labeling_0.4.3            ps_1.7.6                 
[145] plyr_1.8.9                stringi_1.8.4            
[147] deldir_2.0-4              viridisLite_0.4.2        
[149] BiocParallel_1.38.0       assertthat_0.2.1         
[151] munsell_0.5.1             lazyeval_0.2.2           
[153] spatstat.geom_3.2-9       PCAtools_2.16.0          
[155] Matrix_1.7-0              RcppHNSW_0.6.0           
[157] hms_1.1.3                 patchwork_1.2.0          
[159] bit64_4.0.5               sparseMatrixStats_1.16.0 
[161] future_1.33.2             shiny_1.8.1.1            
[163] ROCR_1.0-11               Rfast_2.1.0              
[165] igraph_2.0.3              RcppParallel_5.1.7       
[167] bit_4.0.5                </code></pre>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>