<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>COTAN Datasets analysis - CD14+ clusterizations comparisons with CellTypist</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./NonCT_clusters.html">Cluster evaluation</a></li><li class="breadcrumb-item"><a href="./cd14_clusterizations_comparisons.html">CD14+ clusterizations</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">COTAN Datasets analysis</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Home</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false">
 <span class="menu-text">Datasets</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false">
 <span class="menu-text">Mouse Cortex Loo 2019</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Mouse_Cortex_E14.5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cleaning and cell clustering</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneCoexAnalysisDataset1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Coexpression analaysis</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false">
 <span class="menu-text">Mouse Cortex Yuzwa 2017</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./CorticalCells_GSM2861511_E135-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">E13.5 Cleaning and cell clustering</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneCoexAnalysisE13.5Yuzwa.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Correlation analysis for E13.5</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./CorticalCells_GSM2861514_E175-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">E17.5 Cleaning and cell clustering</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneCoexAnalysisE17.5Yuzwa.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Correlation analysis for E17.5</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="false">
 <span class="menu-text">Mouse Brain LaManno 2021</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ForebrainDorsal_E135-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">E13.5 Cleaning and cell clustering</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneCoexAnalysisE13.5MouseBrain.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Correlation analysis for E13.5</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ForebrainDorsal_E150-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">E15.0 Cleaning and cell clustering</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneCoexAnalysisE15.0MouseBrain.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Correlation analysis for E15.0</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ForebrainDorsal_E175-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">E17.5 Cleaning and cell clustering</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneCoexAnalysisE17.5MouseBrain.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Correlation analysis for E17.5</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ForebrainLaManno2021ClustersGDI.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Original Clusters GDI evaluation</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false">
 <span class="menu-text">CD14+</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./cd14_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CD14+ Cleaning and cell clustering</span></a>
  </div>
</li>
      </ul>
  </li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false">
 <span class="menu-text">PBMC form 10X site</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./PBMC1_filtering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cleaning PBMC1</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./PBMC2_filtering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cleaning PBMC2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./PBMC3_filtering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cleaning PBMC3</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./PBMC4_filtering.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cleaning PBMC4</span></a>
  </div>
</li>
      </ul>
  </li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" aria-expanded="false">
 <span class="menu-text">Coex score evaluation</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-7" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./generalCoexAssess.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">General Coex Evaluation</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" aria-expanded="false">
 <span class="menu-text">GDI sensitivity</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-8" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./OriginalClustersDistanceEvaluation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Clusters Distance Evaluation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./FindUniformOrigCluster.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cluster Unifomity and GDI score</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./GDI_Increment_From_Mixing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">GDI Increment From Mixing</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-9" aria-expanded="true">
 <span class="menu-text">Cluster evaluation</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-9" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-9" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./NonCT_clusters.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Not UT cluster with LR</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Real_UT_clustersByCotan.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Real UT clusters with LR</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./cd14_clusterizations_comparisons.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">CD14+ clusterizations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./clustering_comparison.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PBMCs clustering comparison</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./probability_score.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PBMCs probability score</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-10" aria-expanded="false">
 <span class="menu-text">DEA evaluation</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-10" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-10" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./DEA_Type1Er.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Type I error</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneExpressionAnalysisUniformClusters.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Explorative analysis of gene expression</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./FDR_analisys.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">FDR analysis - define DE genes</span></a>
  </div>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-11" aria-expanded="false">
 <span class="menu-text">FDR and TPR - ROC curves</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-11" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-11" class="collapse list-unstyled sidebar-section depth2 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./FDR_analisysResults3_10.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Results - thresholds true 3% 10%</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./FDR_analisysResults5_20.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Results - thresholds true 5% 20%</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./FDR_analisysResults5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Results - thresholds true 5%</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./FDR_analysisResultsLFC.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Results - Log Fold Change</span></a>
  </div>
</li>
      </ul>
  </li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-12" aria-expanded="false">
 <span class="menu-text">Additional code and scripts</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-12" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-12" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./src/scanpyTypeIError.py" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ScamPy Type I Error code</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./src/scanpyFDR.py" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ScamPy FDR code</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">CD14+ clusterizations comparisons with CellTypist</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">#library(SingleCellExperiment)</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co">#library(DuoClustering2018)</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co">#library(tidyr)</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rlang)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggsankey) <span class="co"># remotes::install_github("davidsjoberg/ggsankey")</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(zeallot)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(assertthat)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(COTAN)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(caret)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_bw</span>())</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(nnet)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="co">#devtools::load_all("~/dev/COTAN/COTAN/")</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">parallelly.fork.enable =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>outDir <span class="ot">&lt;-</span> <span class="st">"Results/Clusterization/"</span></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="fu">setLoggingLevel</span>(<span class="dv">2</span>)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="fu">setLoggingFile</span>(<span class="fu">file.path</span>(outDir, <span class="st">"CD14_Monocytes_ClusterizationsComparisons.log"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>cd14Obj <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="fu">file.path</span>(<span class="st">"Data/CD14Cleaned/"</span>, <span class="st">"CD14_Monocytes.cotan.RDS"</span>))</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>sampleCondition <span class="ot">&lt;-</span> <span class="fu">getMetadataElement</span>(cd14Obj, <span class="fu">datasetTags</span>()[[<span class="st">"cond"</span>]])</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>sampleCondition</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "CD14_Monocytes"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">getClusterizations</span>(cd14Obj)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "split"           "merge"           "majority-voting"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>metaC <span class="ot">&lt;-</span> <span class="fu">getMetadataCells</span>(cd14Obj)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>splitClusters <span class="ot">&lt;-</span> <span class="fu">getClusters</span>(cd14Obj, <span class="st">"split"</span>)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>mergedClusters <span class="ot">&lt;-</span> <span class="fu">getClusters</span>(cd14Obj, <span class="st">"merge"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>labelsDF <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">file.path</span>(<span class="st">"Data/CD14Cleaned/"</span>, <span class="st">"CD14Cleaned_Immune_All_Low_predicted_labels.csv"</span>), <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>labelsDF <span class="ot">&lt;-</span> <span class="fu">column_to_rownames</span>(labelsDF, <span class="at">var =</span> <span class="st">"X"</span>)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(labelsDF) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"[.]"</span>, <span class="st">"-"</span>, <span class="fu">rownames</span>(labelsDF))</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>cells_to_keep <span class="ot">&lt;-</span> <span class="fu">rownames</span>(labelsDF)[<span class="fu">rownames</span>(labelsDF) <span class="sc">%in%</span> <span class="fu">getCells</span>(cd14Obj)]</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="fu">assert_that</span>(<span class="fu">identical</span>(cells_to_keep, <span class="fu">getCells</span>(cd14Obj)))</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>majorityVotingClusters <span class="ot">&lt;-</span> labelsDF[cells_to_keep, <span class="st">"majority_voting"</span>]</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(majorityVotingClusters) <span class="ot">&lt;-</span> cells_to_keep</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>majorityVotingCoexDF <span class="ot">&lt;-</span> <span class="fu">DEAOnClusters</span>(cd14Obj,<span class="at">clName =</span><span class="st">"majority-voting"</span>,<span class="at">clusters =</span>   majorityVotingClusters)</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>cd14Obj <span class="ot">&lt;-</span> <span class="fu">addClusterization</span>(cd14Obj, <span class="at">clName =</span> <span class="st">"majority-voting"</span>,</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>                             <span class="at">clusters =</span> majorityVotingClusters,</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>                             <span class="at">coexDF =</span> majorityVotingCoexDF)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Save the COTAN object</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(cd14Obj, <span class="at">file =</span> <span class="fu">file.path</span>(outDir, <span class="fu">paste0</span>(sampleCondition, <span class="st">".cotan.RDS"</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nlevels</span>(splitClusters)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">sort</span>(<span class="fu">table</span>(splitClusters), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), 10L)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="fu">nlevels</span>(mergedClusters)</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">sort</span>(<span class="fu">table</span>(mergedClusters), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), 10L)</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="fu">nlevels</span>(majorityVotingClusters)</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">sort</span>(<span class="fu">table</span>(majorityVotingClusters), <span class="at">decreasing =</span> <span class="cn">TRUE</span>), 10L)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>splitClustersDF <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(splitClusters)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>splitClustersDF[[<span class="st">"cell"</span>]] <span class="ot">&lt;-</span> <span class="fu">rownames</span>(splitClustersDF)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(splitClustersDF)[[<span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="st">"COTAN.split.cluster"</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>splitClustersDF <span class="ot">&lt;-</span> splitClustersDF[<span class="fu">order</span>(splitClustersDF[[<span class="st">"COTAN.split.cluster"</span>]]), ]</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>mergedClustersDF <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(mergedClusters)</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>mergedClustersDF[[<span class="st">"cell"</span>]] <span class="ot">&lt;-</span> <span class="fu">rownames</span>(mergedClustersDF)</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mergedClustersDF)[[<span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="st">"COTAN.merged.cluster"</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>mergedClustersDF <span class="ot">&lt;-</span> mergedClustersDF[<span class="fu">order</span>(mergedClustersDF[[<span class="st">"COTAN.merged.cluster"</span>]]), ]</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>majorityVotingClustersDF <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(majorityVotingClusters)</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>majorityVotingClustersDF[[<span class="st">"cell"</span>]] <span class="ot">&lt;-</span> <span class="fu">rownames</span>(majorityVotingClustersDF)</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(majorityVotingClustersDF)[[<span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="st">"majority.voting.cluster"</span></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>majorityVotingClustersDF <span class="ot">&lt;-</span> majorityVotingClustersDF[<span class="fu">order</span>(majorityVotingClustersDF[[<span class="st">"majority.voting.cluster"</span>]]), ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>mjvt_split.table <span class="ot">&lt;-</span> <span class="fu">merge.data.frame</span>(<span class="at">x =</span> majorityVotingClustersDF, <span class="at">y =</span> splitClustersDF,</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">by =</span> <span class="st">"cell"</span>, <span class="at">all.x =</span> <span class="cn">TRUE</span>, <span class="at">all.y =</span> <span class="cn">TRUE</span>)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(mjvt_split.table[,<span class="fu">c</span>(2L, 3L)])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                       COTAN.split.cluster
majority.voting.cluster   1   2   3   4   5
    Classical monocytes 877   3 637 837  48
    NK cells              1  25   0   0   0
    pDC                   0  10   0   0   0</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>mjvt_split.table2 <span class="ot">&lt;-</span> mjvt_split.table <span class="sc">%&gt;%</span> <span class="fu">make_long</span>(majority.voting.cluster, COTAN.split.cluster)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(mjvt_split.table2,</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> x,</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">next_x =</span> next_x,</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">node =</span> node,</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">next_node =</span> next_node,</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>           <span class="at">fill =</span> <span class="fu">factor</span>(node),</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> node)) <span class="sc">+</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sankey</span>(<span class="at">flow.alpha =</span> <span class="fl">0.75</span>, <span class="at">node.color =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sankey_label</span>(<span class="at">size =</span> <span class="fl">3.5</span>, <span class="at">color =</span> <span class="dv">1</span>, <span class="at">fill =</span> <span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis_d</span>(<span class="at">option =</span> <span class="st">"A"</span>, <span class="at">alpha =</span> <span class="fl">0.95</span>) <span class="sc">+</span></span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_sankey</span>(<span class="at">base_size =</span> <span class="dv">16</span>) <span class="sc">+</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="cd14_clusterizations_comparisons_files/figure-html/COTAN.split.cluster.sankey.maj.voting-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>mjvt_merged.table <span class="ot">&lt;-</span> <span class="fu">merge.data.frame</span>(<span class="at">x =</span> majorityVotingClustersDF, <span class="at">y =</span> mergedClustersDF,</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">by =</span> <span class="st">"cell"</span>, <span class="at">all.x =</span> <span class="cn">TRUE</span>, <span class="at">all.y =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(mjvt_merged.table[,<span class="fu">c</span>(2L, 3L)])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                       COTAN.merged.cluster
majority.voting.cluster    1    2    3
    Classical monocytes 1522  877    3
    NK cells               0    1   25
    pDC                    0    0   10</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>mjvt_merged.table2 <span class="ot">&lt;-</span> mjvt_merged.table <span class="sc">%&gt;%</span> <span class="fu">make_long</span>(majority.voting.cluster, COTAN.merged.cluster)</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(mjvt_merged.table2,</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> x,</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">next_x =</span> next_x,</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">node =</span> node,</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">next_node =</span> next_node,</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>           <span class="at">fill =</span> <span class="fu">factor</span>(node),</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> node)) <span class="sc">+</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sankey</span>(<span class="at">flow.alpha =</span> <span class="fl">0.75</span>, <span class="at">node.color =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sankey_label</span>(<span class="at">size =</span> <span class="fl">3.5</span>, <span class="at">color =</span> <span class="dv">1</span>, <span class="at">fill =</span> <span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis_d</span>(<span class="at">option =</span> <span class="st">"A"</span>, <span class="at">alpha =</span> <span class="fl">0.95</span>) <span class="sc">+</span></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_sankey</span>(<span class="at">base_size =</span> <span class="dv">16</span>) <span class="sc">+</span></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="cd14_clusterizations_comparisons_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>markersCD14 <span class="ot">&lt;-</span> <span class="fu">findClustersMarkers</span>(cd14Obj,<span class="at">n =</span> <span class="dv">100</span>,<span class="at">clName =</span> <span class="st">"merge"</span>,<span class="at">method =</span> <span class="st">"BH"</span>)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(markersCD14,<span class="at">file =</span> <span class="st">"Data/CD14Cleaned/ClusterMarkerGenes.csv"</span>)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(markersCD14)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  CL     Gene      Score      adjPVal         DEA IsMarker  logFoldCh
1  1 HLA-DQA1 -0.3398915 2.603676e-59 -0.15274999        0 -0.5470222
2  1 HLA-DQA2 -0.2899185 7.002901e-43 -0.12223685        0 -0.4804835
3  1 HLA-DPA1 -0.2808830 2.589504e-40 -0.13090249        0 -0.4370669
4  1  HLA-DRA -0.2499997 1.042557e-31 -0.10296185        0 -0.3654458
5  1 HLA-DPB1 -0.2225778 5.652934e-25 -0.09887797        0 -0.3375752
6  1     YBX1 -0.2095002 5.042102e-22 -0.08080263        0 -0.3784618</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>cd14GDI <span class="ot">&lt;-</span> <span class="fu">calculateGDI</span>(cd14Obj)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>subsetGDI <span class="ot">&lt;-</span> cd14GDI[cd14GDI<span class="sc">$</span>sum.raw.norm <span class="sc">&gt;</span> <span class="dv">7</span>,]</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>top.GDI.genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(subsetGDI[<span class="fu">order</span>(subsetGDI<span class="sc">$</span>GDI,<span class="at">decreasing =</span> T),])[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>]</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="fu">GDIPlot</span>(cd14Obj,<span class="at">genes =</span> <span class="st">""</span>,<span class="at">GDIIn =</span> cd14GDI)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="cd14_clusterizations_comparisons_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">getNormalizedData</span>(cd14Obj)</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> data[<span class="sc">!</span><span class="fu">rowSums</span>(<span class="fu">as.matrix</span>(data)) <span class="sc">&lt;</span> <span class="dv">1</span>,]</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">log</span>(data<span class="sc">*</span><span class="dv">10000</span><span class="sc">+</span><span class="dv">1</span>)</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>row_stdev <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">1</span>, sd, <span class="at">na.rm=</span><span class="cn">TRUE</span>)</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>row_stdev <span class="ot">&lt;-</span> row_stdev[<span class="fu">order</span>(row_stdev,<span class="at">decreasing =</span> T)]</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>genes.to.keep <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">names</span>(row_stdev[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>]),top.GDI.genes)</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>data.small <span class="ot">&lt;-</span> data[<span class="fu">rownames</span>(data) <span class="sc">%in%</span> genes.to.keep,]</span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a><span class="co">#data &lt;- t(as.matrix(data))</span></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>data.small <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">as.matrix</span>(data.small))</span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a>COTAN_Cl.code <span class="ot">&lt;-</span> <span class="fu">getClusterizationData</span>(cd14Obj,<span class="at">clName =</span> <span class="st">"merge"</span>)[[<span class="dv">1</span>]]</span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a><span class="co">#COTAN_Cl.code[COTAN_Cl.code == "A549"] &lt;- 0</span></span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a><span class="co">#COTAN_Cl.code[COTAN_Cl.code == "CCL-185-IG"] &lt;- 1</span></span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a>COTAN_Cl.code <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(COTAN_Cl.code)</span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a>data.small <span class="ot">&lt;-</span> <span class="fu">cbind</span>(data.small,COTAN_Cl.code)</span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a>data.small <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(data.small)</span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-26"><a href="#cb21-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Split the data into training and test set</span></span>
<span id="cb21-27"><a href="#cb21-27" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb21-28"><a href="#cb21-28" aria-hidden="true" tabindex="-1"></a>training.samples <span class="ot">&lt;-</span> data.small[,<span class="st">"COTAN_Cl.code"</span>] <span class="sc">%&gt;%</span> </span>
<span id="cb21-29"><a href="#cb21-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">createDataPartition</span>(<span class="at">p =</span> <span class="fl">0.8</span>, <span class="at">list =</span> <span class="cn">FALSE</span>)</span>
<span id="cb21-30"><a href="#cb21-30" aria-hidden="true" tabindex="-1"></a>train.data  <span class="ot">&lt;-</span> data.small[training.samples, ]</span>
<span id="cb21-31"><a href="#cb21-31" aria-hidden="true" tabindex="-1"></a>test.data <span class="ot">&lt;-</span> data.small[<span class="sc">-</span>training.samples, ]</span>
<span id="cb21-32"><a href="#cb21-32" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(train.data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                    RPL22   LAPTM5 ATP6V0B      RPL5      MCL1      CTSS
AAACATACCACTAG-1 9.485384 9.485384       0  0.000000  0.000000  0.000000
AAACATACGTTCAG-1 0.000000 0.000000       0  0.000000 10.114628  0.000000
AAACATTGCTTCGC-1 0.000000 0.000000       0  0.000000  0.000000  0.000000
AAACATTGGGCAAG-1 9.322304 0.000000       0  8.629246  9.727739  0.000000
AAACGGCTACGGAG-1 0.000000 0.000000       0 10.582413  9.483852 10.870089
AAACGGCTAGTCAC-1 0.000000 9.124915       0  9.124915  0.000000  9.818008
                   S100A10   S100A11    S100A9    S100A8    S100A6   S100A4
AAACATACCACTAG-1  9.485384  0.000000  9.485384  9.485384  9.485384 11.56476
AAACATACGTTCAG-1  0.000000 10.114628 11.906354 11.367363 10.807755 11.36736
AAACATTGCTTCGC-1  9.414354 10.512912 11.493724 11.812175 11.206045 11.71687
AAACATTGGGCAAG-1 10.238541  9.322304 10.826311 11.519449 10.420856 11.33713
AAACGGCTACGGAG-1 10.176961  0.000000 12.191831 11.881678 10.176961 11.56323
AAACGGCTAGTCAC-1  0.000000 10.511128 12.120544 11.763871 10.916584 11.20426
                     VAMP8     COX5B   NBEAL1 EEF1B2     ARPC2     RPL15
AAACATACCACTAG-1 10.583946  0.000000 9.485384      0 10.178493 11.094761
AAACATACGTTCAG-1 10.114628  0.000000 0.000000      0  0.000000  9.421522
AAACATTGCTTCGC-1  9.414354  0.000000 9.414354      0  9.414354  9.414354
AAACATTGGGCAAG-1  9.727739  8.629246 9.322304      0  8.629246 11.113989
AAACGGCTACGGAG-1 10.176961 10.176961 0.000000      0  9.483852 10.582413
AAACGGCTAGTCAC-1  9.818008  9.818008 9.124915      0  9.818008 10.916584
                      RPSA      GPX1     RPL29    SERP1      BTF3      CD14
AAACATACCACTAG-1  0.000000  9.485384  0.000000 0.000000  0.000000 10.178493
AAACATACGTTCAG-1  9.421522  9.421522  0.000000 0.000000 10.114628 10.520080
AAACATTGCTTCGC-1  0.000000  9.414354 10.512912 0.000000  0.000000  0.000000
AAACATTGGGCAAG-1 10.931670 10.420856 10.015406 9.322304  0.000000  0.000000
AAACGGCTACGGAG-1  0.000000 11.429697  9.483852 0.000000  0.000000  9.483852
AAACGGCTAGTCAC-1  0.000000 10.223455  9.124915 9.124915  9.124915  0.000000
                    DUSP1   HIGD2A  PRELID1     HLA-E      AIF1    CLIC1
AAACATACCACTAG-1 0.000000 9.485384 0.000000  0.000000  9.485384 10.58395
AAACATACGTTCAG-1 9.421522 0.000000 0.000000  0.000000  9.421522  0.00000
AAACATTGCTTCGC-1 9.414354 9.414354 0.000000  0.000000 11.206045  0.00000
AAACATTGGGCAAG-1 9.322304 9.322304 9.727739  8.629246  9.727739  0.00000
AAACGGCTACGGAG-1 0.000000 0.000000 0.000000  0.000000  0.000000  0.00000
AAACGGCTAGTCAC-1 9.124915 0.000000 9.124915 10.511128  9.818008  0.00000
                   HLA-DRA  HLA-DRB5  HLA-DRB1 HLA-DQA1  HLA-DQB1  HLA-DMA
AAACATACCACTAG-1 10.583946 10.871621 11.883210        0 10.178493 0.000000
AAACATACGTTCAG-1  9.421522  0.000000  9.421522        0  0.000000 0.000000
AAACATTGCTTCGC-1  0.000000 10.107460 10.512912        0  0.000000 9.414354
AAACATTGGGCAAG-1 10.575003 10.015406 10.575003        0  9.727739 0.000000
AAACGGCTACGGAG-1  0.000000  9.483852  9.483852        0  0.000000 0.000000
AAACGGCTAGTCAC-1 10.223455  0.000000  9.818008        0  9.124915 0.000000
                 HLA-DPA1  HLA-DPB1    RPS10    SF3B5  RNASET2      RAC1
AAACATACCACTAG-1 9.485384  9.485384 0.000000 0.000000 0.000000 10.178493
AAACATACGTTCAG-1 0.000000  0.000000 9.421522 0.000000 0.000000  0.000000
AAACATTGCTTCGC-1 0.000000  9.414354 0.000000 0.000000 0.000000  9.414354
AAACATTGGGCAAG-1 0.000000 11.113989 9.727739 9.727739 8.629246  0.000000
AAACGGCTACGGAG-1 0.000000  0.000000 0.000000 0.000000 9.483852  0.000000
AAACGGCTAGTCAC-1 0.000000  0.000000 9.124915 0.000000 0.000000  9.124915
                     TOMM7     PPIA CHCHD2    ARPC1B TMEM176B   SLC25A6 TIMP1
AAACATACCACTAG-1  9.485384 0.000000      0  9.485384 0.000000 10.178493     0
AAACATACGTTCAG-1 10.114628 9.421522      0 10.114628 0.000000 10.114628     0
AAACATTGCTTCGC-1 10.107460 0.000000      0  0.000000 9.414354 10.512912     0
AAACATTGGGCAAG-1  8.629246 0.000000      0  8.629246 0.000000  8.629246     0
AAACGGCTACGGAG-1  0.000000 0.000000      0 10.176961 9.483852  0.000000     0
AAACGGCTAGTCAC-1  0.000000 0.000000      0  9.818008 9.124915  9.124915     0
                      CFP     CEBPD   PABPC1     EDF1      KLF6       VIM
AAACATACCACTAG-1  0.00000  0.000000 0.000000 9.485384 10.178493 10.178493
AAACATACGTTCAG-1 10.11463  0.000000 0.000000 0.000000 10.114628  9.421522
AAACATTGCTTCGC-1  0.00000  9.414354 0.000000 0.000000  9.414354  0.000000
AAACATTGGGCAAG-1 10.70853  8.629246 0.000000 9.322304 10.015406  9.727739
AAACGGCTACGGAG-1  0.00000 11.093229 0.000000 0.000000 10.176961  0.000000
AAACGGCTAGTCAC-1  0.00000 10.223455 9.818008 0.000000  0.000000  9.818008
                     SRGN   TALDO1     LSP1   MS4A6A    NEAT1    ATP5L     TPI1
AAACATACCACTAG-1 9.485384 0.000000 9.485384 9.485384 0.000000 0.000000 0.000000
AAACATACGTTCAG-1 0.000000 9.421522 0.000000 0.000000 9.421522 0.000000 0.000000
AAACATTGCTTCGC-1 0.000000 0.000000 0.000000 0.000000 9.414354 9.414354 0.000000
AAACATTGGGCAAG-1 8.629246 0.000000 9.727739 0.000000 9.322304 0.000000 8.629246
AAACGGCTACGGAG-1 9.483852 0.000000 0.000000 0.000000 0.000000 0.000000 0.000000
AAACGGCTAGTCAC-1 9.124915 9.124915 0.000000 9.818008 0.000000 0.000000 0.000000
                  ARHGDIB     CD63      NACA LINC00936     ARPC3   ATP5EP2
AAACATACCACTAG-1 0.000000 0.000000  0.000000  0.000000 10.178493  0.000000
AAACATACGTTCAG-1 0.000000 9.421522  9.421522 10.114628  0.000000  9.421522
AAACATTGCTTCGC-1 9.414354 0.000000  0.000000  0.000000  9.414354 10.512912
AAACATTGGGCAAG-1 8.629246 8.629246  9.727739  8.629246  0.000000  8.629246
AAACGGCTACGGAG-1 0.000000 0.000000 10.176961  9.483852  9.483852  0.000000
AAACGGCTAGTCAC-1 9.124915 9.124915  0.000000  0.000000  9.818008  9.124915
                     TPT1    PSME1     PSME2   NFKBIA   RPL36AL     NPC2
AAACATACCACTAG-1 0.000000 0.000000 10.178493 0.000000  0.000000 9.485384
AAACATACGTTCAG-1 9.421522 9.421522  0.000000 0.000000  9.421522 0.000000
AAACATTGCTTCGC-1 9.414354 0.000000  0.000000 0.000000  0.000000 0.000000
AAACATTGGGCAAG-1 0.000000 8.629246  9.322304 9.322304  9.322304 0.000000
AAACGGCTACGGAG-1 0.000000 0.000000  9.483852 0.000000 10.176961 0.000000
AAACGGCTAGTCAC-1 9.818008 0.000000  9.818008 9.124915  9.124915 9.818008
                    CRIP1    ANXA2     RPL4    TCEB2   CORO1A    PYCARD
AAACATACCACTAG-1  0.00000 0.000000 10.58395 0.000000 9.485384 10.871621
AAACATACGTTCAG-1  0.00000 9.421522 10.11463 9.421522 0.000000  0.000000
AAACATTGCTTCGC-1  0.00000 0.000000  0.00000 0.000000 9.414354  9.414354
AAACATTGGGCAAG-1 10.42086 9.322304  0.00000 0.000000 8.629246  9.727739
AAACGGCTACGGAG-1  0.00000 9.483852  0.00000 0.000000 0.000000 10.176961
AAACGGCTAGTCAC-1  0.00000 0.000000  0.00000 0.000000 9.124915  9.818008
                    COTL1     APRT   GABARAP    EIF4A1 UBB     RPL23 GRN
AAACATACCACTAG-1 0.000000 0.000000  9.485384  0.000000   0  0.000000   0
AAACATACGTTCAG-1 0.000000 0.000000  9.421522  0.000000   0  9.421522   0
AAACATTGCTTCGC-1 0.000000 0.000000  0.000000 10.107460   0  9.414354   0
AAACATTGGGCAAG-1 0.000000 8.629246  8.629246  9.322304   0 10.015406   0
AAACGGCTACGGAG-1 0.000000 0.000000 10.176961  0.000000   0  0.000000   0
AAACGGCTAGTCAC-1 9.124915 0.000000  9.124915  0.000000   0  0.000000   0
                     RPL38    ACTG1    SCAND1    PSMA7     RPS21       CFD
AAACATACCACTAG-1  0.000000 0.000000  9.485384  0.00000  0.000000  9.485384
AAACATACGTTCAG-1 10.114628 0.000000  0.000000  0.00000  9.421522  0.000000
AAACATTGCTTCGC-1  0.000000 0.000000  0.000000 10.51291 10.107460 10.512912
AAACATTGGGCAAG-1  8.629246 9.322304  0.000000 10.01541  8.629246  0.000000
AAACGGCTACGGAG-1  0.000000 0.000000 10.176961  0.00000  9.483852  0.000000
AAACGGCTAGTCAC-1  0.000000 9.124915  9.818008 10.22345  9.818008  9.124915
                     GPX4     ATP5D  UQCR11.1  NDUFA11 C19orf43     JUNB
AAACATACCACTAG-1 11.88321  0.000000  0.000000 0.000000 9.485384 0.000000
AAACATACGTTCAG-1  0.00000  0.000000 10.114628 9.421522 0.000000 0.000000
AAACATTGCTTCGC-1  0.00000  0.000000  0.000000 0.000000 0.000000 9.414354
AAACATTGGGCAAG-1  0.00000  8.629246  9.727739 9.322304 8.629246 0.000000
AAACGGCTACGGAG-1  0.00000 10.176961  0.000000 9.483852 0.000000 0.000000
AAACGGCTAGTCAC-1  0.00000  0.000000  0.000000 9.818008 0.000000 0.000000
                      IER2     PGLS      JUND   NDUFA13     FXYD5   COX6B1
AAACATACCACTAG-1  0.000000 0.000000  9.485384  0.000000  0.000000 9.485384
AAACATACGTTCAG-1 11.030895 0.000000  0.000000 10.114628  0.000000 9.421522
AAACATTGCTTCGC-1  9.414354 0.000000  9.414354 10.107460 10.107460 0.000000
AAACATTGGGCAAG-1  0.000000 8.629246  8.629246  0.000000  9.322304 8.629246
AAACGGCTACGGAG-1  0.000000 0.000000 10.176961  9.483852  0.000000 9.483852
AAACGGCTAGTCAC-1 11.427402 0.000000  9.124915  9.124915  0.000000 9.818008
                     HCST    TYROBP      GMFG    ZFP36    BLVRB     FOSB
AAACATACCACTAG-1 0.000000 10.178493  0.000000 0.000000 0.000000 0.000000
AAACATACGTTCAG-1 0.000000 10.807755 10.114628 0.000000 0.000000 9.421522
AAACATTGCTTCGC-1 0.000000 10.800587  9.414354 0.000000 0.000000 9.414354
AAACATTGGGCAAG-1 8.629246 10.575003  9.322304 0.000000 9.322304 0.000000
AAACGGCTACGGAG-1 0.000000  9.483852  0.000000 0.000000 9.483852 0.000000
AAACGGCTAGTCAC-1 0.000000 11.522711  9.124915 9.818008 9.124915 9.818008
                     AP2S1   GLTSCR2      EMP3     FCGRT      RPS5    LGALS2
AAACATACCACTAG-1  0.000000  9.485384  9.485384  0.000000  9.485384 10.178493
AAACATACGTTCAG-1  0.000000  9.421522  0.000000  0.000000  0.000000  0.000000
AAACATTGCTTCGC-1  0.000000  0.000000  9.414354  0.000000 10.107460  0.000000
AAACATTGGGCAAG-1  8.629246  0.000000 10.826311  0.000000 10.015406  9.727739
AAACGGCTACGGAG-1 10.176961  0.000000 10.870089 10.176961 10.176961  0.000000
AAACGGCTAGTCAC-1  0.000000 10.223455  0.000000  9.124915  0.000000  0.000000
                   MT-ND2  MT-ATP6   MT-ND4   MT-CYB COTAN_Cl.code
AAACATACCACTAG-1 0.000000 0.000000 9.485384 0.000000             1
AAACATACGTTCAG-1 0.000000 9.421522 0.000000 9.421522             1
AAACATTGCTTCGC-1 0.000000 9.414354 0.000000 9.414354             1
AAACATTGGGCAAG-1 8.629246 9.727739 0.000000 8.629246             2
AAACGGCTACGGAG-1 0.000000 0.000000 0.000000 9.483852             1
AAACGGCTAGTCAC-1 9.124915 9.124915 0.000000 0.000000             1</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>train.data<span class="sc">$</span>COTAN_Cl.code <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(train.data<span class="sc">$</span>COTAN_Cl.code)</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>test.data<span class="sc">$</span>COTAN_Cl.code <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(test.data<span class="sc">$</span>COTAN_Cl.code)</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Fit the model</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="co">#model &lt;- glm( COTAN_Cl.code ~., data = train.data, family = binomial,control = list(maxit = 75))</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="fu">multinom</span>(COTAN_Cl.code <span class="sc">~</span> ., <span class="at">data =</span> train.data, <span class="at">maxit =</span> <span class="dv">500</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># weights:  369 (244 variable)
initial  value 2143.392575 
iter  10 value 761.340646
iter  20 value 746.268326
iter  30 value 720.679315
iter  40 value 579.207155
iter  50 value 545.792232
iter  60 value 539.711792
iter  70 value 526.902032
iter  80 value 506.380523
iter  90 value 485.108188
iter 100 value 480.679083
iter 110 value 476.395523
iter 120 value 462.778421
iter 130 value 457.035815
iter 140 value 455.285417
iter 150 value 449.873877
iter 160 value 446.861152
iter 170 value 444.452737
iter 180 value 439.972413
iter 190 value 437.990151
iter 200 value 437.057149
iter 210 value 436.908590
iter 220 value 436.792687
iter 230 value 436.667356
iter 240 value 436.372853
iter 250 value 435.784259
iter 260 value 435.132550
iter 270 value 434.964836
iter 280 value 434.951070
iter 290 value 434.949618
iter 300 value 434.949374
final  value 434.949335 
converged</code></pre>
</div>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>probabilities <span class="ot">&lt;-</span> <span class="fu">predict</span>(model, <span class="at">newdata =</span> test.data, <span class="at">type =</span> <span class="st">"probs"</span>)</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Find the class with the highest probability for each case</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>predicted.classes <span class="ot">&lt;-</span> <span class="fu">apply</span>(probabilities, <span class="dv">1</span>, which.max)</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Adjust predicted classes to match your factor levels</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>levels <span class="ot">&lt;-</span> <span class="fu">levels</span>(train.data<span class="sc">$</span>COTAN_Cl.code)</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>predicted.classes <span class="ot">&lt;-</span> levels[predicted.classes]</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate model accuracy</span></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>accuracy <span class="ot">&lt;-</span> <span class="fu">mean</span>(predicted.classes <span class="sc">==</span> test.data<span class="sc">$</span>COTAN_Cl.code)</span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a><span class="co">#result.df[nrow(result.df),"Accuracy"] &lt;- accuracy</span></span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(accuracy)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.8521561</code></pre>
</div>
</div>
<p>So the logistic regression accuracy for COTAN three clusters is quite good.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Data/CD14Cleaned/FromStringrEnrichr/CD14Cluster1.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Cluster 1 top enriched genes. In red enriched Reactome Pathways innate immune system genes.</figcaption>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Data/CD14Cleaned/FromStringrEnrichr/CD14Cluster2string_vector_graphic.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">cluster 2 - 100 top enriched genes. In red Antigen processing and presentation of exogenous peptide antigen via MHC class II genes, in violet Regulation of leukocyte cell-cell adhesion (GO Biological Process) and in green Adaptive Immune System</figcaption>
</figure>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Data/CD14Cleaned/FromStringrEnrichr/CD14Cluster3.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Cluster 3 - 100 top enriched genes. In violet Cytokine Signaling in Immune system and in red Immunoregulatory interactions between a Lymphoid and a non-Lymphoid cell enriched genes for Reatome Pathways.</figcaption>
</figure>
</div>
<p>The cells in cluster 1 seem to be classical CD14 monocytes since they express CD14 while depleted in MHC class II proteins complex which is enriched in cell cluster 2 that seems intermediate monocytes.</p>
<p>Cluster 3 does not seem to be a monocyte cluster. Using enrichr website it is enriched in Plasmacytoid Dendritic cell marker genes.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 20.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       
 [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   
 [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          
[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   

time zone: Europe/Rome
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] nnet_7.3-19        stringr_1.5.1      caret_6.0-94       lattice_0.22-6    
 [5] COTAN_2.5.2        assertthat_0.2.1   zeallot_0.1.0      tibble_3.2.1      
 [9] ggsankey_0.0.99999 ggplot2_3.5.1      rlang_1.1.3       

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.22          splines_4.4.0            
  [3] later_1.3.2               polyclip_1.10-6          
  [5] hardhat_1.3.1             pROC_1.18.5              
  [7] rpart_4.1.23              fastDummies_1.7.3        
  [9] lifecycle_1.0.4           doParallel_1.0.17        
 [11] globals_0.16.3            MASS_7.3-60.2            
 [13] dendextend_1.17.1         magrittr_2.0.3           
 [15] plotly_4.10.4             rmarkdown_2.27           
 [17] yaml_2.3.8                httpuv_1.6.15            
 [19] Seurat_5.1.0              sctransform_0.4.1        
 [21] spam_2.10-0               askpass_1.2.0            
 [23] sp_2.1-4                  spatstat.sparse_3.0-3    
 [25] reticulate_1.37.0         cowplot_1.1.3            
 [27] pbapply_1.7-2             RColorBrewer_1.1-3       
 [29] lubridate_1.9.3           abind_1.4-5              
 [31] zlibbioc_1.50.0           Rtsne_0.17               
 [33] purrr_1.0.2               BiocGenerics_0.50.0      
 [35] ipred_0.9-14              lava_1.8.0               
 [37] circlize_0.4.16           IRanges_2.38.0           
 [39] S4Vectors_0.42.0          ggrepel_0.9.5            
 [41] irlba_2.3.5.1             listenv_0.9.1            
 [43] spatstat.utils_3.0-4      umap_0.2.10.0            
 [45] goftest_1.2-3             RSpectra_0.16-1          
 [47] spatstat.random_3.2-3     dqrng_0.4.0              
 [49] fitdistrplus_1.1-11       parallelly_1.37.1        
 [51] DelayedMatrixStats_1.26.0 leiden_0.4.3.1           
 [53] codetools_0.2-20          DelayedArray_0.30.1      
 [55] tidyselect_1.2.1          shape_1.4.6.1            
 [57] farver_2.1.2              ScaledMatrix_1.12.0      
 [59] viridis_0.6.5             matrixStats_1.3.0        
 [61] stats4_4.4.0              spatstat.explore_3.2-7   
 [63] jsonlite_1.8.8            GetoptLong_1.0.5         
 [65] progressr_0.14.0          ggridges_0.5.6           
 [67] survival_3.6-4            iterators_1.0.14         
 [69] foreach_1.5.2             tools_4.4.0              
 [71] ica_1.0-3                 Rcpp_1.0.12              
 [73] glue_1.7.0                prodlim_2023.08.28       
 [75] gridExtra_2.3             SparseArray_1.4.5        
 [77] xfun_0.44                 MatrixGenerics_1.16.0    
 [79] ggthemes_5.1.0            dplyr_1.1.4              
 [81] withr_3.0.0               fastmap_1.2.0            
 [83] fansi_1.0.6               openssl_2.2.0            
 [85] digest_0.6.35             rsvd_1.0.5               
 [87] timechange_0.3.0          parallelDist_0.2.6       
 [89] R6_2.5.1                  mime_0.12                
 [91] colorspace_2.1-0          scattermore_1.2          
 [93] tensor_1.5                spatstat.data_3.0-4      
 [95] utf8_1.2.4                tidyr_1.3.1              
 [97] generics_0.1.3            data.table_1.15.4        
 [99] recipes_1.0.10            class_7.3-22             
[101] httr_1.4.7                htmlwidgets_1.6.4        
[103] S4Arrays_1.4.1            ModelMetrics_1.2.2.2     
[105] uwot_0.2.2                pkgconfig_2.0.3          
[107] gtable_0.3.5              timeDate_4032.109        
[109] ComplexHeatmap_2.20.0     lmtest_0.9-40            
[111] XVector_0.44.0            htmltools_0.5.8.1        
[113] dotCall64_1.1-1           clue_0.3-65              
[115] SeuratObject_5.0.2        scales_1.3.0             
[117] png_0.1-8                 gower_1.0.1              
[119] knitr_1.46                rstudioapi_0.16.0        
[121] reshape2_1.4.4            rjson_0.2.21             
[123] nlme_3.1-164              zoo_1.8-12               
[125] GlobalOptions_0.1.2       KernSmooth_2.23-24       
[127] parallel_4.4.0            miniUI_0.1.1.1           
[129] RcppZiggurat_0.1.6        pillar_1.9.0             
[131] grid_4.4.0                vctrs_0.6.5              
[133] RANN_2.6.1                promises_1.3.0           
[135] BiocSingular_1.20.0       beachmat_2.20.0          
[137] xtable_1.8-4              cluster_2.1.6            
[139] evaluate_0.23             cli_3.6.2                
[141] compiler_4.4.0            crayon_1.5.2             
[143] future.apply_1.11.2       labeling_0.4.3           
[145] plyr_1.8.9                stringi_1.8.4            
[147] viridisLite_0.4.2         deldir_2.0-4             
[149] BiocParallel_1.38.0       munsell_0.5.1            
[151] lazyeval_0.2.2            spatstat.geom_3.2-9      
[153] PCAtools_2.16.0           Matrix_1.7-0             
[155] RcppHNSW_0.6.0            patchwork_1.2.0          
[157] sparseMatrixStats_1.16.0  future_1.33.2            
[159] shiny_1.8.1.1             ROCR_1.0-11              
[161] Rfast_2.1.0               igraph_2.0.3             
[163] RcppParallel_5.1.7       </code></pre>
</div>
</div>



</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>